--- deadwood-2.9.03/src/DwUdpSocket.c	2010-08-06 09:41:58.000000000 -0700
+++ deadwood-2.9.04/src/DwUdpSocket.c	2010-08-06 16:00:10.000000000 -0700
@@ -630,11 +630,22 @@
 
         local_id = get_dns_qid(packet, len, 2);
         if(local_id == -2) { /* Immediate NOTIMPL for EDNS requests w/ OPT */
+#ifdef STRICT_RFC2671_COMPLIANCE
                 packet[2] = 0x81; /* QR = 1; Op = 0; AA = 0; TC = 0; RD = 1 */
                 packet[3] = 0x84; /* RA = 1; Z = 0; RCODE = "notimpl" (4) */
                 sendto(sock,(void *)packet,len,0,(struct sockaddr *)&client,
                                 c_len);
                 goto catch_get_local_udp_packet;
+#else /* STRICT_RFC2671_COMPLIANCE */
+		query = dw_get_dname_type(packet,12,len);
+		if(query == 0) {
+                	goto catch_get_local_udp_packet;
+		}
+		len = query->len + 14;
+		packet[11] = 0; /* We no longer have final question */
+		dw_destroy(query);
+        	local_id = get_dns_qid(packet, len, 2);
+#endif /* STRICT_RFC2671_COMPLIANCE */
         }
         if(local_id < 0 || len < 13) { /* Invalid remote packet */
                 goto catch_get_local_udp_packet;
