--- deadwood-2.9.03/doc/CHANGELOG	2010-08-06 09:41:58.000000000 -0700
+++ deadwood-2.9.04/doc/CHANGELOG	2010-08-13 01:47:05.000000000 -0700
@@ -1,3 +1,18 @@
+2.9.04 (20100813):
+
+* Updated EDNS handling to treat EDNS packets as if the OPT record does not
+  exist and we're getting a RFC1035 DNS packet.  If the behavior of returning
+  NOTIMPL to EDNS packets is preferred, this can be enabled by defining
+  "STRICT_RFC2671_COMPLIANCE" when compiling Deadwood.
+
+* DNS queries are now case-insensitive, with case preserved for the query
+  originally sent to Deadwood (just in case a stub resolver violates RFC1035
+  and is case sensitive)
+
+* Default timeout_seconds is now 2, no longer 3
+
+* The highest maxprocs is now 16,384, not 1,024
+
 2.9.03 (20100806):
 
 * Updated dwood2rc_n_timeout_seconds SQA test to work around CentOS bug
