--- maradns-1.4.01/doc/en/text/Makefile	2007-10-19 08:54:14.000000000 -0500
+++ maradns-1.4.02/doc/en/text/Makefile	2009-12-28 16:12:55.000000000 -0600
@@ -101,6 +101,6 @@
 	../../../tools/ej/ej2txt ../source/troubleshoot.ej > troubleshoot.txt
 
 coding_style.txt: ../source/coding_style.ej
-	../../../tools/ej/ej2html ../source/coding_style.ej > \
+	../../../tools/ej/ej2txt ../source/coding_style.ej > \
 	    coding_style.txt
 
--- maradns-1.4.01/doc/en/source/coding_style.ej	2007-10-19 08:54:14.000000000 -0500
+++ maradns-1.4.02/doc/en/source/coding_style.ej	2009-12-28 16:18:06.000000000 -0600
@@ -4,7 +4,7 @@
 
 </HEAD><BODY>
 
-<!-- Copyright 2007 Sam Trenholme
+<!-- Copyright 2007,2009 Sam Trenholme
 
     TERMS
 
@@ -95,13 +95,13 @@
 I went to a good deal of effort to make sure over 99% of MaraDNS 1.2 
 zone files work completely unchanged in MaraDNS 1.3, and that a MaraDNS 
 user can set a single mararc variable, <tt>csv2_tilde_handling</tt> in 
-order to make MaraDNS 1.3 100% compatible with MaraDNS 1.2 zone files.
+order to make MaraDNS 1.4 100% compatible with MaraDNS 1.2 zone files.
 <p>
 
-Indeed, MaraDNS 1.3 still supports MaraDNS 1.0 zone files, even though 
+Indeed, MaraDNS 1.4 still supports MaraDNS 1.0 zone files, even though 
 I am encouraging people to upgrade to the newer zone file format, and 
 even though I have a Perl script that converts a MaraDNS 1.0 zone file 
-in to a MaraDNS 1.2 zone file.
+in to a MaraDNS 1.2+ zone file.
 <p>
 
 I have sometimes sometimes made old configuration files not compatible 
@@ -137,6 +137,12 @@
 
 There's also the issue, with French users of MaraDNS, of the French 
 documentation not being updated since the 1.0 days of MaraDNS.
+<p>
+
+Note that MaraDNS 2.0 will break many configurations, since the same
+IP will be unable to be both a recursive and authoritative DNS server.
+This is why I will continue to support MaraDNS 1.4 after when and if
+MaraDNS 2.0 comes out.
 
 <h2>Brace and tab style</h2>
 
@@ -169,7 +175,7 @@
 which is a BSD license as follows:
 
 <blockquote>
-Copyright (c) 2002-2007 Sam Trenholme
+Copyright (c) 2002-2009 Sam Trenholme and others
 <p>
 TERMS
 <p>
