--- maradns-1.4.01/sqa/regressions/README	2007-10-19 08:54:14.000000000 -0500
+++ maradns-1.4.02/sqa/regressions/README	2010-01-07 11:31:08.000000000 -0600
@@ -1,8 +1,8 @@
 This directory contains various tests, mainly ones that detect the 
 presence of known MaraDNS bugs that have been fixed.
 
-All of these tests can be run at once by typing in 'sh ./do.tests' from this
-directory.
+All of these tests can be run at once by typing in 'sh ./do.tests' as
+root from this directory after compiling MaraDNS.
 
 These tests can also be run by typing in 'sh ./do.test' in the appropriate
 directory for the test in question.  MaraDNS needs to be compiled before
