--- deadwood-3.2.02/src/DwRecurse.c	2012-02-29 01:59:53.000000000 -0600
+++ deadwood-3.2.03/src/DwRecurse.c	2012-07-21 13:07:03.000000000 -0500
@@ -1868,8 +1868,9 @@
         }
 
         /* First CNAME pointer... */
-        value = dwx_make_one_cname_rr(query, 0, action, 2, ttl, &len, 768);
-        if(value == 0) {
+        value = dwx_make_one_cname_rr(query, 0, action, 2, ttl, &len, 1024);
+        if(value == 0 || value->len > 1020) {
+		dw_log_string("dwx_make_one_cname_rr problem",100);
                 goto catch_dwx_create_cname_reply;
         }
         offsets[0] = 0;
@@ -1890,7 +1891,8 @@
                 }
                 offsets[this_offset * 2] = value->len;
                 offsets[(this_offset * 2) + 1] = value->len + len;
-                if(dw_append(temp, value) == -1) {
+                if(dw_append(temp, value) == -1 || value->len > 1020) {
+			dw_log_string("Append error making cname reply",100);
                         goto catch_dwx_create_cname_reply;
                 }
                 dw_destroy(temp);
@@ -1901,6 +1903,11 @@
         /* Final answer after CNAME pointers... */
         dwx_add_final_answer(query, answer, value, offsets, this_offset);
 
+	if(value->len > 1020) {
+		dw_log_string("Final append error making cname reply",100);
+		goto catch_dwx_create_cname_reply;
+	}
+
         /* Make that a full answer to give the calling function */
         return value;
 
--- deadwood-3.2.03/sqa/do.tests.orig	2012-07-21 13:20:33.000000000 -0500
+++ deadwood-3.2.03/sqa/do.tests	2012-07-21 13:21:13.000000000 -0500
@@ -91,6 +91,7 @@
 TESTS=$TESTS"dwood2rc_d_upstream_servers "
 TESTS=$TESTS"dwood2rc_n_verbose_level "
 TESTS=$TESTS"recurse_2link_cname "
+TESTS=$TESTS"recurse_es-us.noticias.yahoo.com "
 TESTS=$TESTS"sqa_badid_question "
 TESTS=$TESTS"sqa_bigpacket " # Needs BIND to run
 TESTS=$TESTS"sqa_cname_chain "
