--- deadwood-3.2.02/sqa/do.tests	2012-02-11 14:15:43.000000000 -0600
+++ deadwood-3.2.03/sqa/do.tests	2012-04-23 13:40:48.000000000 -0500
@@ -51,6 +51,7 @@
 TESTS=$TESTS"dwood2rc_n_maxprocs,2 "
 TESTS=$TESTS"dwood2rc_s_cache_file,2 "
 TESTS=$TESTS"sqa_one_upstream_down,2 "
+TESTS=$TESTS"sqa_easydns_bad_truncation,2 "
 TESTS=$TESTS"dwood2rc_n_num_retries,2 "
 
 # These tests are stable and don't fail
