--- deadwood-3.2.02/sqa/dwood2rc_d_upstream_servers/do.test	2010-01-19 08:25:48.000000000 -0800
+++ deadwood-3.2.03/sqa/dwood2rc_d_upstream_servers/do.test	2012-12-20 10:29:11.478377209 -0800
@@ -43,13 +43,13 @@
 
 ../../src/Deadwood -f dwood2rc | grep -v version | sed 's/[(].*[)]//' &
 sleep 1
-askmara Awww1.example.com. 127.0.0.1
+askmara Awww1.example.com. 127.0.0.1 | sed 's/86399/86400/'
 echo
-askmara Awww1.example.com. 127.0.0.1
+askmara Awww1.example.com. 127.0.0.1 | sed 's/86399/86400/'
 echo
-askmara Awww2.example.com. 127.0.0.1
+askmara Awww2.example.com. 127.0.0.1 | sed 's/86399/86400/'
 echo
-askmara Awww1.example.net. 127.0.0.1
+askmara Awww1.example.net. 127.0.0.1 | sed 's/86399/86400/'
 
 killall maradns
 killall Deadwood
