--- deadwood-3.2.03/src/DwSys.c.orig	2012-09-22 11:00:56.118909325 -0500
+++ deadwood-3.2.03/src/DwSys.c	2012-09-22 11:04:32.292701151 -0500
@@ -631,7 +631,8 @@
 
 /* Drop privileges and become unprivileged user */
 void sandbox() {
-#if ! (defined MINGW || defined __CYGWIN__)
+#ifndef MINGW 
+#ifndef __CYGWIN__
         unsigned char *c = 0;
         gid_t g = DW_UID;
         if(key_s[DWM_S_chroot_dir] == 0) {
@@ -642,13 +643,14 @@
                 dw_fatal("Converting chroot_dir to string failed");
         }
         if(chdir((char *)c) != 0) {
+                printf("There is no directory %s\n",(char *)c);
                 dw_fatal("chdir() failed");
         }
 #ifndef QNX
         if(chroot((char *)c) == -1) {
                 dw_fatal("chroot() failed");
         }
-#endif
+#endif /* QNX */
         if(setgroups(1,&g) == -1) {
                 dw_fatal("setgroups() failed");
         }
@@ -667,6 +669,7 @@
                 c = 0;
         }
         return;
+#endif /* __CYGWIN__ */
 #endif /* MINGW */
 }
 
