--- maradns-1.2.01/doc/en/source/convert.ej	2005-12-02 15:10:29.000000000 -0800
+++ maradns-1.2.02/doc/en/source/convert.ej	2006-01-02 08:00:37.000000000 -0800
@@ -4,7 +4,7 @@
 
 </HEAD><BODY>
 
-<!-- Copyright 2005 Sam Trenholme
+<!-- Copyright 2005-2006 Sam Trenholme
 
     TERMS
 
@@ -59,5 +59,24 @@
 </blockquote>
 </ul>
 
-</body>
+<hr>
+
+If one is converting from another DNS server, it is probably best to have
+MaraDNS act in a strictly RFC-compliant manner.  In particular, star
+records should probably be handled in the exact same manner described
+in section 4.3.3 of RFC1034, and MaraDNS should be configured to have
+full support for DNS over TCP.
+
+<p>
 
+To make star record handling strictly RFC compliant, add the following line
+to the <tt>mararc</tt> configuration file:
+
+<blockquote>
+<tt>bind_star_handling = 1</tt>
+</blockquote>
+
+To give MaraDNS full DNS-over-TCP support, follow the directions in the 
+<A href=dnstcp.html>dnstcp</A> document.
+
+</body>
--- maradns-1.2.01/doc/en/source/copy	2005-10-02 17:47:05.000000000 -0700
+++ maradns-1.2.02/doc/en/source/copy	2006-01-02 07:37:44.000000000 -0800
@@ -1,4 +1,4 @@
-<!-- Copyright 2005 Sam Trenholme
+<!-- Copyright 2005-2006 Sam Trenholme
 
     TERMS
 
--- maradns-1.2.01/doc/en/source/SOURCE.FILES	2002-03-24 01:58:18.000000000 -0800
+++ maradns-1.2.02/doc/en/source/SOURCE.FILES	2006-01-02 07:55:27.000000000 -0800
@@ -4,13 +4,26 @@
 changelog.embed: MaraDNS changelog
 compile.ej: Tutorial on compiling MaraDNS
 convert.ej: Tutorial on moving from another DNS server to MaraDNS
+copy: The copyright statement for MaraDNS documents
 credits.ej: Wrapper for credits.embed
 credits.embed: People who have helped make MaraDNS a reality
-csv1.ej: Csv1(5) man page (describes MaraDNS' zone file format)
+csv1.ej: Csv1(5) man page (describes MaraDNS' legacy zone file format)
+csv2.ej: Csv2(5) man page (describes MaraDNS' zone file format)
+csv2_txt.ej: Csv2_txt(5) man page (describes TXT and "RAW" RRs in csv2 zones)
+dangling.ej: How to handling the "dangling CNAME record" warning
 data_structures.ej: Technical notes on MaraDNS' internal data structures
+default_zonefile.ej: How to have stars at the end of hostnames
+dns_intro.ej: Description of DNS for newbies (currently unused; to be merged)
+dnsintro.ej: DNS introductory document used in MaraDNS tutorial
+dnsmaster.ej: How to allow others to transfer zones from MaraDNS
+dnsslave.ej: How to transfer zones from other servers to MaraDNS
+dnstcp.ej: How to perform DNS over TCP with MaraDNS
+download.ej: Old webpage template for MaraDNS download page (to be deleted)
+duende.ej: Man page for Duende MaraDNS daemonizer
+ej.doc.format: Description of EJ's document format
 ej.template: Tamplate for making new ej files
 example_auth_mararc: ASCII example authoritative-only mararc file
-example_csv1: ASCII example csv1 zone file
+example_csv2: ASCII example csv2 zone file
 example_full_mararc: ASCII mararc example showing most mararc parameters
 example_mararc: ASCII default simplified example mararc file
 example_mararc.ej: ej wrapper for example_mararc
@@ -18,16 +31,19 @@
 example_simple_csv1: ASCII example simplified csv1 zone file
 faq.ej: ej wrapper for faq.embed
 faq.embed: MaraDNS FAQ
+fetchzone.ej: Man page for fetchzone zone transfer program
 files.ej: List of files in the top-level MaraDNS directory
-getzone.ej: Man page for getzone
+getzone.ej: Man page for legacy csv1 getzone program
 glossary.ej: Glossary of DNS terms for the tutorial
 maradns.ej: MaraDNS man page
 mararc.ej: mararc(5) man page; described the MaraDNS man page format
+old.changelog: Old change entries (to be deleted; info already merged)
 quick_start.ej: Quick start guide for MaraDNS
 recordtypes.ej: Description of some DNS record types in the tutorial
 recursive.ej: Tutorial on setting up a recursive DNS server
 security_design.embed: Description of the MaraDNS security design
 SOURCE.FILES: This file; describes files in this directory
-tuorial.ej: Beginning of tutorial
+tutorial.ej: Index of tutorial files
+update.ej: How to update from MaraDNS 1.0 to MaraDNS 1.2
 zoneserver.ej: Man page for the zoneserver
 
