--- deadwood-2.9.05/doc/CHANGELOG	2010-08-13 01:48:00.000000000 -0700
+++ deadwood-2.9.06/doc/CHANGELOG	2010-09-03 07:03:11.000000000 -0700
@@ -1,3 +1,36 @@
+2.9.06 (20100903):
+
+* Documentation updates
+
+* Some unused library routines trimmed from Deadwood via OTHER_STUFF
+  compile-time flag
+
+* All references to obsolete deliver_all variable removed from source code
+
+* Tuned handling of empty DNS packets: Empty DNS packets are now treated
+  like low-TTL "not there" replies unless the RCODE is "SERVER FAIL"; in
+  that case, empty DNS packets are ignored.
+
+2.9.05 (20100820):
+
+* Workaround for broken behavior with www.bookride.com's DNS server where
+  a CNAME referral is incorrectly marked as a NXDOMAIN
+
+* ANY queries that point to CNAMEs now work in Deadwood
+
+* check_ip_acl speeded up by breaking out of its loop when we get past
+  the last user-defined ACL
+
+* mkSecretTxt file added to Windows port of Deadwood ; its source is in
+  tools/
+
+* doc/internals updated for Deadwood 2.9 
+
+* compile.options now lists all #ifdef flags in Deadwood's source
+
+* INSTALL.txt updated to point out we are now using MinGW 3.4.2; 
+  Makefile.mingw310 now called Makefile.mingw342
+
 2.9.04 (20100813):
 
 * Updated EDNS handling to treat EDNS packets as if the OPT record does not
