--- deadwood-3.0.05/doc/CHANGELOG	2011-11-10 11:46:48.000000000 -0600
+++ deadwood-3.1.01/doc/CHANGELOG	2011-11-27 15:33:46.484939409 -0600
@@ -1,3 +1,24 @@
+3.1.01 (20111127):
+
+* Bugfix: We now favor glued over glueless records so as to not fall in
+  to a "tarpit" where we keep chasing the same glueless records over and 
+  over.
+
+* Bugfix: Truncation hack now correctly sets NSCOUNT and ARCOUNT to 0.
+  This speeds up resolving yahoo.com hosts.
+
+* Code size optimization: reset_rem() is now a function instead of an
+  inline.  Reduces -Os size of 32-bit Windows binary by some 3k.
+
+* Code size optimization: Optional TINY_BINARY compile flag which makes
+  the binary a little smaller by removing some of the parsing rules.
+
+* Resolution speed optimization: We now look for a cached incomplete
+  CNAME record and will use it if available.  This should speed up 
+  resolution when there is a high-TTL incomplete CNAME record pointing 
+  to a low-TTL record.  As it turns out, this update doesn't change the
+  binary format for the cache file.
+
 3.0.05 (20111110):
 
 * Bugfix: reject_ptr now works
