--- deadwood-3.0.01/src/DwDnsStr.c	2010-09-24 18:45:12.000000000 -0700
+++ deadwood-3.0.02/src/DwDnsStr.c	2010-11-07 12:20:27.000000000 -0800
@@ -517,6 +517,18 @@
                 if((ipc & 0xff000000) == 0x0a000000) { /* 10.x.x.x */
                         return 1;
                 }
+                if((ipc & 0xff000000) == 0x7f000000) { /* 127.x.x.x */
+                        return 1;
+                }
+                if((ipc & 0xffff0000) == 0xa9fe0000) { /* 169.254.x.x */
+                        return 1;
+		}
+                if((ipc & 0xff000000) == 0xe0000000) { /* 224.x.x.x */
+                        return 1;
+                }
+                if((ipc & 0xffff0000) == 0x00000000) { /* 0.0.x.x */
+                        return 1;
+                }
         }
 
         if(blacklist_hash == 0) {
--- deadwood-3.0.01/doc/Deadwood.ej	2010-09-24 18:45:11.000000000 -0700
+++ deadwood-3.0.02/doc/Deadwood.ej	2010-11-07 12:29:43.000000000 -0800
@@ -85,9 +85,36 @@
 port 53
 
 <h2>filter_rfc1918</h2>
-When this is set to 1, RFC1918 (private) IP addresses are not allowed to be
-in DNS replies.  See http://crypto.stanford.edu/dns/ for why this improves
-security.
+When this has a value of 1, a number of different IP ranges are not allowed
+to be in DNS A replies:
+
+<ul>
+<li>192.168.x.x
+<li>172.[16-31].x.x
+<li>10.x.x.x
+<li>127.x.x.x
+<li>169.254.x.x
+<li>224.x.x.x
+<li>0.0.x.x
+</ul>
+
+If one of the above IPs is detected in a DNS reply, and filter_rfc1918 has
+a value of 1, Deadwood will return a synthetic "this host does not reply"
+response (a SOA record in the NS section) instead of the A record.
+
+<p>
+
+The reason for this is to provide a "dnswall" that protects users for some
+kinds of attacks, as described at http://crypto.stanford.edu/dns/
+
+<p>
+
+Please note that Deadwood only provides IPv4 "dnswall" functionality and
+does not help protect against IPv6 answers.  If protection against certain
+IPv6 AAAA records is needed, either disable all AAAA answers by setting
+reject_aaaa to have a value of 1, or use an external program to filter
+undesired IPv4 answers (such as the dnswall program).
+
 <p>
 The default value for this is 1
 
diff -u deadwood-3.0.01/sqa/recurse_2link_cname/do.test deadwood-3.0.02/sqa/recurse_2link_cname/do.test
--- deadwood-3.0.01/sqa/recurse_2link_cname/do.test	2010-07-15 21:21:31.000000000 -0600
+++ deadwood-3.0.02/sqa/recurse_2link_cname/do.test	2011-01-28 21:22:58.000000000 -0700
@@ -21,6 +21,7 @@
 chroot_dir="/etc/deadwood"
 root_servers = {}
 root_servers["."] = "127.0.0.2"
+filter_rfc1918 = 0
 EOF
 
 ../../src/Deadwood -f dwood3rc > /dev/null 2>&1 &
