--- deadwood-3.0.01/doc/CHANGELOG	2010-09-24 19:45:11.000000000 -0600
+++ deadwood-3.0.02/doc/CHANGELOG	2011-01-28 18:37:20.000000000 -0700
@@ -1,4 +1,13 @@
-3.0.01 (201010??):
+3.0.02 (20110128):
+
+* Bugfix: "www.urbandictionary.com" resolves again
+
+* Bugfix: filter_rfc1918 parameter now filters more IPv4 DNS answers that
+  might be exploitable.
+
+* Third party feature: Yarin has added "--pid" support to "duende"
+
+3.0.01 (20100924):
 
 * dwood3rc parser fixes: Dictionary elements now must be initialized; 
   referring non-existent dictionary elements is a fatal parse error; 
--- deadwood-3.0.01/doc/Windows/dwood3rc.txt	2010-09-24 19:45:11.000000000 -0600
+++ deadwood-3.0.02/doc/Windows/dwood3rc.txt	2011-01-28 18:35:09.000000000 -0700
@@ -2,7 +2,7 @@
 # text file parser.
 
 # If you wish to use this program to cache from other recursive servers
-# instead of doing its own recursion, uncomment the following line
+# instead of doing its own recursion, uncomment the following lines
 #upstream_servers = {}
 #upstream_servers["."]="8.8.8.8, 8.8.4.4" # Servers we connect to
 
