--- deadwood-3.0.01/src/DwHash.c	2010-09-24 19:45:12.000000000 -0600
+++ deadwood-3.0.02/src/DwHash.c	2011-01-29 10:06:53.000000000 -0700
@@ -517,11 +517,11 @@
                 new = 0;
         }
         if(key_copy != 0) {
-                free(key_copy);
+                dw_destroy(key_copy);
                 key_copy = 0;
         }
         if(val_copy != 0) {
-                free(val_copy);
+                dw_destroy(val_copy);
                 val_copy = 0;
         }
         return 0;
--- deadwood-3.0.01/sqa/do.tests	2010-09-08 22:54:51.000000000 -0600
+++ deadwood-3.0.02/sqa/do.tests	2011-01-29 10:09:02.000000000 -0700
@@ -94,6 +94,13 @@
 TESTS=$TESTS"sqa_roy_arends "
 TESTS=$TESTS"sqa_tcp_buffering "
 # This test is last since it recompiles Deadwood
+# We run it seven times because some memleaks only happen some of the time
+TESTS=$TESTS"sqa_valgrind "
+TESTS=$TESTS"sqa_valgrind "
+TESTS=$TESTS"sqa_valgrind "
+TESTS=$TESTS"sqa_valgrind "
+TESTS=$TESTS"sqa_valgrind "
+TESTS=$TESTS"sqa_valgrind "
 TESTS=$TESTS"sqa_valgrind "
 
 for PAIR in $TESTS ; do
