--- maradns-1.2.07.4/build/maradns-1.2.07.4.spec	2006-05-22 23:13:13.000000000 -0700
+++ maradns-1.2.07.5/build/maradns-1.2.07.5.spec	2006-05-29 00:52:45.000000000 -0700
@@ -1,12 +1,12 @@
 Summary: An authoritative and recursive DNS server made with security in mind
 Name: maradns
-Version: 1.2.07.4
+Version: 1.2.07.5
 Release: 1
 License: BSD (Two-clause)
 Group: Networking/Daemons
-Source: http://www.maradns.org/download/1.2/1.2.07.4/maradns-1.2.07.4.tar.bz2
-Source1: maradns-1.2.07.4.tar.bz2.sha1.asc
-Source2: maradns-1.2.07.4.tar.bz2.rmd.asc
+Source: http://www.maradns.org/download/1.2/1.2.07.5/maradns-1.2.07.5.tar.bz2
+Source1: maradns-1.2.07.5.tar.bz2.sha1.asc
+Source2: maradns-1.2.07.5.tar.bz2.rmd.asc
 Patch0: maradns-1.1.59-rpm.patch
 BuildRoot: /var/tmp/%{name}-buildroot
 
@@ -49,7 +49,7 @@
 
 %files
 %defattr(-,root,root)
-%doc /usr/doc/maradns-1.2.07.4/*
+%doc /usr/doc/maradns-1.2.07.5/*
 
 /usr/sbin/maradns
 /usr/sbin/zoneserver
@@ -88,6 +88,9 @@
 echo MaraDNS should have been stopped
 
 %changelog
+* Mon May 29 2006 Sam Trenholme <sam+i55e6bt@chaosring.org>
+- MaraDNS rpm package updated for version 1.2.07.5
+
 * Tue May 16 2006 Sam Trenholme <sam+i55e6bt@chaosring.org>
 - MaraDNS rpm package updated for version 1.2.07.4
 
