--- maradns-1.2.07.4/doc/en/webpage/download.embed.orig	2006-05-23 00:54:40.000000000 -0700
+++ maradns-1.2.07.4/doc/en/webpage/download.embed	2006-05-23 11:55:21.000000000 -0700
@@ -3,20 +3,20 @@
 <h1>&iexcl;Download MaraDNS!</h1>
 
 <h2>Stable release</h2>
-<p>This is the current stable release of MaraDNS, MaraDNS 1.2.07.3
+<p>This is the current stable release of MaraDNS, MaraDNS 1.2.07.5
 <p>This release is available both at <A 
 href="http://sourceforge.net/project/showfiles.php?group_id=24033&amp;package_id=16396">sourceforge.net</A>
 and on this page.
 
 <p><A 
-href="download/1.2/1.2.07.3/maradns-1.2.07.3.tar.bz2"><!--
-                         -->maradns-1.2.07.3.tar.bz2</A>
+href="download/1.2/1.2.07.5/maradns-1.2.07.5.tar.bz2"><!--
+                         -->maradns-1.2.07.5.tar.bz2</A>
 <p><A 
-href="download/1.2/1.2.07.3/maradns-1.2.07.3.tar.bz2.sha1.asc">GPG signature</A><p>
+href="download/1.2/1.2.07.5/maradns-1.2.07.5.tar.bz2.sha1.asc">GPG signature</A><p>
 <p><A 
-href="download/1.2/1.2.07.3/maradns-1.2.07.3.tar.bz2.rmd.asc">GPG signature
+href="download/1.2/1.2.07.5/maradns-1.2.07.5.tar.bz2.rmd.asc">GPG signature
 using RIPEMD160 as the digest algorithm</A><p>
-(released 2006/05/15)
+(released 2006/05/29)
 
 <p>
 The following are files compressed with LZMA compression, which both compresses
@@ -25,35 +25,35 @@
 href="download/1.2/00Uncompress.txt">here</A>.
 
 <p><A 
-href="download/1.2/1.2.07.3/maradns-1.2.07.3.tar.lzma"><!--
-                         -->maradns-1.2.07.3.tar.lzma</A>
+href="download/1.2/1.2.07.5/maradns-1.2.07.5.tar.lzma"><!--
+                         -->maradns-1.2.07.5.tar.lzma</A>
 <p><A 
-href="download/1.2/1.2.07.3/maradns-1.2.07.3.tar.lzma.sha1.asc">GPG signature</A><p>
+href="download/1.2/1.2.07.5/maradns-1.2.07.5.tar.lzma.sha1.asc">GPG signature</A><p>
 <p><A 
-href="download/1.2/1.2.07.3/maradns-1.2.07.3.tar.lzma.rmd.asc">GPG signature
+href="download/1.2/1.2.07.5/maradns-1.2.07.5.tar.lzma.rmd.asc">GPG signature
 using RIPEMD160 as the digest algorithm</A><p>
 
 <p>
 For people who prefer a gzip compressed archive:
 
 <p><A 
-href="download/1.2/1.2.07.3/maradns-1.2.07.3.tar.gz"><!--
-                         -->maradns-1.2.07.3.tar.gz</A>
+href="download/1.2/1.2.07.5/maradns-1.2.07.5.tar.gz"><!--
+                         -->maradns-1.2.07.5.tar.gz</A>
 <p><A 
-href="download/1.2/1.2.07.3/maradns-1.2.07.3.tar.gz.sha1.asc">GPG signature</A><p>
+href="download/1.2/1.2.07.5/maradns-1.2.07.5.tar.gz.sha1.asc">GPG signature</A><p>
 <p><A 
-href="download/1.2/1.2.07.3/maradns-1.2.07.3.tar.gz.rmd.asc">GPG signature
+href="download/1.2/1.2.07.5/maradns-1.2.07.5.tar.gz.rmd.asc">GPG signature
 using RIPEMD160 as the digest algorithm</A><p>
 
 <h3>RPM files</h3>
-<p><A href="download/1.2/1.2.07.3/maradns-1.2.07.3-1.src.rpm"><!--
-                               -->maradns-1.2.07.3-1.src.rpm</A></p>
+<p><A href="download/1.2/1.2.07.5/maradns-1.2.07.5-1.src.rpm"><!--
+                               -->maradns-1.2.07.5-1.src.rpm</A></p>
 <p><i>Greg Swallow fixed the rpm package so that the RPM now builds on
    CentOS3 and CentOS4, in addition to RedHat 6.2.</i>
-<p><b>Binary RPM</b> (Built on RedHat 6.2; verified to work in newer Linuxes)
+<p><b>Binary RPM</b> (Built on CentOS 3.7; verified to work in newer Linuxes)
 </p>
-<p><A href="download/1.2/1.2.07.3/maradns-1.2.07.3-1.i386.rpm"><!--
-                               -->maradns-1.2.07.3-1.i386.rpm</A></p>
+<p><A href="download/1.2/1.2.07.5/maradns-1.2.07.5-1.i386.rpm"><!--
+                               -->maradns-1.2.07.5-1.i386.rpm</A></p>
 
 <h3>Windows Binary</h3>
 <p>I have made a partial port of MaraDNS to Windows.  While this port
@@ -65,19 +65,19 @@
 <A href="http://www.cygwin.com">Cygwin</A>.
 
 <p><A 
-href="download/1.2/1.2.07.3/maradns-1-2-07-3-win32.zip"><!--
-                         -->maradns-1-2-07-3-win32.zip</A>
+href="download/1.2/1.2.07.5/maradns-1-2-07-5-win32.zip"><!--
+                         -->maradns-1-2-07-5-win32.zip</A>
 <p><A 
-href="download/1.2/1.2.07.3/maradns-1-2-07-3-win32.zip.sha1.asc">GPG signature</A><p>
+href="download/1.2/1.2.07.5/maradns-1-2-07-5-win32.zip.sha1.asc">GPG signature</A><p>
 <p><A 
-href="download/1.2/1.2.07.3/maradns-1-2-07-3-win32.zip.rmd.asc">GPG signature
+href="download/1.2/1.2.07.5/maradns-1-2-07-5-win32.zip.rmd.asc">GPG signature
 using RIPEMD160 as the digest algorithm</A><p>
 
 <h3>Slackware Package</h3>
 
 Daniel Zilli has made a Slackware package of maradns, which can be
 downloaded <A
-href="download/1.2/3rd_party/maradns-1.2.07.3-i486-2.tgz">here</A>.<p>
+href="download/1.2/3rd_party/maradns-1.2.07.4-i486-2.tgz">here</A>.<p>
 
 <hr>
 
