--- maradns-1.2.03.1/build/Makefile.linux	2006-01-02 08:59:33.000000000 -0800
+++ maradns-1.2.04/build/Makefile.linux	2006-02-02 12:20:57.000000000 -0800
@@ -29,7 +29,7 @@
 # Debug
 D=$(M) DEBUG=-DDEBUG
 
-FLAGS = -O2 -Wall
+FLAGS = -O2 -Wall -DSELECT_PROBLEM
 M="CC=$(CC) $(FLAGS)"
 #FLAGS = -g
 
@@ -37,7 +37,7 @@
 	cd libs ; make $(M) ; cd ../dns ; make $(M) ; \
 	cd ../rng ; make $(M) ; cd ../parse ; make $(M) ; \
 	cd ../qual ; make $(M) ; cd ../server ; \
-	make $(M) $(Q) $(V) COMPILED=\"$(COMPILED)\" ; \
+	make $(M) $(V) COMPILED=\"$(COMPILED)\" ; \
 	cd ../tools ; make $(M) ; \
 	cd ../tcp ; make $(M) $(V) ; cat ../00README.FIRST
 
