--- maradns-1.2.12.02/dns/Compress.c	2006-08-09 18:14:58.000000000 -0500
+++ maradns-compress_style/dns/Compress.c	2006-08-09 18:14:40.000000000 -0500
@@ -536,6 +536,11 @@
             /* Text data fields */
             else if(subtype == RRSUB_TEXT) {
 	        int len;
+		if(state->uncompressed_offset >= 
+		   state->uncompressed->unit_count) {
+		    state->valid_state = 0;
+                    return JS_ERROR;
+                    }
                 len = *(state->uncompressed->string + 
 		        state->uncompressed_offset);
                 if(len < 0 || len > 256) {
