--- maradns-1.2.12.01/sqa/franky.hosts.test	2005-11-17 04:02:21.000000000 -0600
+++ maradns-1.2.12.02/sqa/franky.hosts.test	2006-08-05 22:54:01.000000000 -0500
@@ -6,6 +6,7 @@
 
 NUM=1
 ASKMARA=../tools/$ASKMARA
+HOSTFILE=hosts3
 
 while : ; do
 	echo Test round $NUM
@@ -15,7 +16,8 @@
 	echo >> sqa.output
 	date >> sqa.output
 	ps auxw | grep maradns >> sqa.output
-	cat ~/mara/franky-test/hosts3 | sed 's/[^A-Za-z0-9\-\.\n]//g' | awk '
+	cat $HOSTFILE | sed 's/[^A-Za-z0-9\-\.\n]//g' | awk '
+            BEGIN{srand();}
 	    { a++;
             if(a % 7 == 0) {
 		    print "./sendudp Awww.google.com. 127.0.0.4 > /dev/null"}
@@ -25,17 +27,24 @@
 		    print "./sendudp Awww.msn.com. 127.0.0.4 > /dev/null"}
             if(a % 83 == 0) {
 		    print "./sendudp Awww.myspace.com. 127.0.0.4 > /dev/null"}
+	    if(a % 17 == 0) {
+            w = int(rand() * 256);
+            x = int(rand() * 256);
+            y = int(rand() * 256);
+            z = int(rand() * 256);
+ print "./sendudp P" w "." x "." y "." z ".in-addr.arpa. 127.0.0.4 > /dev/null"
+               }
 	    print "./sendudp A"$0". 127.0.0.4 > /dev/null"}' | sh
 	sleep 10
         # Make sure we'll still able to resolve host names
-	$ASKMARA 1:www.google.com. >> sqa.output
-	$ASKMARA 1:www.yahoo.com. >> sqa.output
-	$ASKMARA 1:www.msn.com. >> sqa.output
-	$ASKMARA 1:www.microsoft.com. >> sqa.output
-	$ASKMARA 1:www.redhat.com. >> sqa.output
-	$ASKMARA 1:www.distrowatch.com. >> sqa.output
-	$ASKMARA 1:www.myspace.com. >> sqa.output
-	$ASKMARA 1:www.maradns.org. >> sqa.output
-	$ASKMARA 1:www.samiam.org. >> sqa.output
+	$ASKMARA 1:www.google.com. 127.0.0.4 >> sqa.output
+	$ASKMARA 1:www.yahoo.com. 127.0.0.4 >> sqa.output
+	$ASKMARA 1:www.msn.com. 127.0.0.4 >> sqa.output
+	$ASKMARA 1:www.microsoft.com. 127.0.0.4 >> sqa.output
+	$ASKMARA 1:www.redhat.com. 127.0.0.4 >> sqa.output
+	$ASKMARA 1:www.distrowatch.com. 127.0.0.4 >> sqa.output
+	$ASKMARA 1:www.myspace.com. 127.0.0.4 >> sqa.output
+	$ASKMARA 1:www.maradns.org. 127.0.0.4 >> sqa.output
+	$ASKMARA 1:www.samiam.org. 127.0.0.4 >> sqa.output
 done
 
