--- maradns-1.2.12.01/build/maradns-1.2.12.01.spec	2006-07-26 10:58:44.000000000 -0500
+++ maradns-1.2.12.02/build/maradns-1.2.12.02.spec	2006-08-14 14:57:08.000000000 -0500
@@ -1,12 +1,12 @@
 Summary: An authoritative and recursive DNS server made with security in mind
 Name: maradns
-Version: 1.2.12.01
+Version: 1.2.12.02
 Release: 1
 License: BSD (Two-clause)
 Group: Networking/Daemons
-Source: http://www.maradns.org/download/1.2/1.2.12.01/maradns-1.2.12.01.tar.bz2
-Source1: maradns-1.2.12.01.tar.bz2.sha1.asc
-Source2: maradns-1.2.12.01.tar.bz2.rmd.asc
+Source: http://www.maradns.org/download/1.2/1.2.12.02/maradns-1.2.12.02.tar.bz2
+Source1: maradns-1.2.12.02.tar.bz2.sha1.asc
+Source2: maradns-1.2.12.02.tar.bz2.rmd.asc
 Patch0: maradns-1.1.59-rpm.patch
 BuildRoot: /var/tmp/%{name}-buildroot
 
@@ -49,7 +49,7 @@
 
 %files
 %defattr(-,root,root)
-%doc /usr/doc/maradns-1.2.12.01/*
+%doc /usr/doc/maradns-1.2.12.02/*
 
 /usr/sbin/maradns
 /usr/sbin/zoneserver
@@ -107,6 +107,9 @@
 fi
 
 %changelog
+* Fri Jul 28 2006 Sam Trenholme <sam+i55e6bt@chaosring.org>
+- MaraDNS rpm package updated for version 1.2.12.02
+
 * Tue Jul 25 2006 Sam Trenholme <sam+i55e6bt@chaosring.org>
 - MaraDNS rpm package updated for version 1.2.12.01
 
