--- deadwood-3.1.02/src/DwRecurse.c	2011-11-27 20:30:56.000000000 -0600
+++ deadwood-3.1.03/src/DwRecurse.c	2011-11-30 10:48:26.725523737 -0600
@@ -2005,16 +2005,6 @@
                 goto catch_dwx_make_cname_reply;
         }
 
-        uncomp = dwx_create_cname_reply(query, action, answer, 60);
-        comp = dwc_compress(query, uncomp);
-
-        /* Send a reply out for each "live" local connection */
-        if(uncomp == 0 || comp == 0 || rem[conn_num].local == 0 ||
-           rem[conn_num].num_locals < 1) {
-                goto catch_dwx_make_cname_reply;
-        }
-
-        /* Put answer in cache as full answer */
         ttl = dw_get_a_dnsttl(answer,0,31536000,1);
         if(ttl < 0) {
                 dw_log_dwstr("Invalid TTL in answer ",answer,100);
@@ -2024,6 +2014,16 @@
                 ttl = 60;
         }
         /*ttl = 30; // DEBUG*/
+        uncomp = dwx_create_cname_reply(query, action, answer, ttl);
+        comp = dwc_compress(query, uncomp);
+
+        /* Send a reply out for each "live" local connection */
+        if(uncomp == 0 || comp == 0 || rem[conn_num].local == 0 ||
+           rem[conn_num].num_locals < 1) {
+                goto catch_dwx_make_cname_reply;
+        }
+
+        /* Put answer in cache as full answer */
         dwh_add(cache, query, uncomp, ttl, 1);
 
         /* Send answer from cache */
