--- deadwood-3.2.03/doc/CHANGELOG	2012-02-28 23:59:53.000000000 -0800
+++ deadwood-3.2.04/doc/CHANGELOG	2013-12-20 07:47:13.000000000 -0800
@@ -1,10 +1,98 @@
-3.2.02 (2012????):
+3.2.04 (20131220):
 
-* New dwood3rc parameter added: max_ttl
+- Security patch as described at http://samiam.org/blog/20131202.html
+
+- One line patch so that whatever.scalzi.com resolves again.  More 
+  information at http://cct2.vk.tj 
+
+- Seven-line patch so mixed-case names correctly resolve.
+
+- Fairly large patch to both speed up some DNS resolutions and to get
+  oncetv-ipn.net to resolve again
+
+- Added FAQ entry about Deadwood dropping RFC1918 addresses from replies
+
+- Added FAQ entry about what causes "uninitialized dictionary
+  variable" message
+
+- Exit with fatal error when there is over 20,000 upstream/root server
+  entries in dwood3rc (older Deadwood releases would silently not work)
+
+- Compression is now case insensitive (smaller DNS packets)
+
+- Expanded recursive_acl example to show multiple subnets
+
+- Fixed warnings when compiled with clang 3.0
+
+- d.root-servers.org moved from 128.8.10.90 to 199.7.91.13
+
+- Cleanup of code in now-case-insensitive dwx_dname_issame()
+
+- Cleanup of code that initializes "dns string" objects; make
+  sure initialized values are 0
+
+3.2.03d (20131202):
+
+- Security patch as described at http://samiam.org/blog/20131202.html
+
+3.2.03c (20130713):
+
+- Seven-line patch so mixed-case names correctly resolve.
+
+3.2.03b (20130501):
+
+- Deadwood compiled in Windows with -Os and stripped
 
-* "Recursive-algorithm" document added
+3.2.03a (20130423):
+
+- One line patch so that whatever.scalzi.com resolves again.  More 
+  information at http://cct2.vk.tj
+
+3.2.03 (20121220):
+
+- Added a whole bunch of security validation to DwCompress.c (always
+  make sure offsets are within bounds)
+
+- Deadwood now compiles with IPv6 support again
+
+- We now handle EasyDNS' bad truncation in a reasonable manner
+
+- Added new SQA test for es-us.noticias.yahoo.com issue in May/June 2012
+
+- Replaced "malloc" with "dw_malloc" wrapper (make it a little easier
+  for embedded devs)
+
+- Updated INSTALL.txt (Windows 7; Deadwood's malloc use)
+
+- There is now a compile-time flag (-DSHOWPACKET) to see every single
+  packet Deadwood receives (for debugging)
+
+- If /etc/deadwood is missing, we now tell them what the missing directory is
+
+- Made the underlying RNG a little faster and about 50 bytes smaller
+  (I like keeping the Windows binary under 65,536 bytes in size)
+
+- Documented the difference between a string, numeric, and dictionary parameter
+
+- If the Deadwood cache file is older than the dwood3rc file, do not
+  load the cache
+
+- SQA update: Netstat changed, breaking one of the SQA tests. These
+  SQA tests have been updated to pass again (and should work when run
+  against an older netstat)
+
+- SQA update: Sometimes the ttl ages one second, which made one of the
+  tests sometimes fail.
+
+- OS Update: The supported OSes for Deadwood are now CentOS/RedHat 6 and
+  Windows 7
+
+3.2.02 (20120229):
+
+* New dwood3rc parameter added: max_ttl
 
-* INSTALL.TXT updated
+* Documentation updates: "Recursive-algorithm" document added, INSTALL.TXT
+  updated, etc.
 
 3.2.01 (20120211):
 
