--- deadwood-3.2.04/src/DwRecurse.c.orig	2013-03-16 01:21:04.000000000 -0700
+++ deadwood-3.2.04/src/DwRecurse.c	2013-03-16 01:21:17.000000000 -0700
@@ -2018,7 +2018,7 @@
                 goto catch_dwx_make_cname_reply;
         }
 
-        ancount = dw_fetch_u16(answer,-7);
+        ancount = dw_fetch_u16(answer,-6);
         if(ancount < 1) {
                 goto catch_dwx_make_cname_reply;
         }
