--- deadwood-3.2.03/doc/dwood3rc-all	2012-02-28 23:59:53.000000000 -0800
+++ deadwood-3.2.04/doc/dwood3rc-all	2013-10-28 10:55:10.807099126 -0700
@@ -83,7 +83,9 @@
 recurse_number_ports = 4096
 
 # Who is allowed to use the cache
-recursive_acl = "127.0.0.1/16" 
+recursive_acl =  "127.0.0.1/16"
+recursive_acl += ", 10.1.0.0/16" 
+recursive_acl += ", 192.168.1.0/24" 
 
 # Whether to not resolve AAAA (IPv6) queries.  This should be 1 on IPv4-only
 # networks; default is 0
