--- deadwood-2.9.07/tools/ej/ej2html	2005-12-06 01:43:38.000000000 -0800
+++ deadwood-3.0.01/tools/ej/ej2html	2010-09-21 10:54:47.000000000 -0700
@@ -15,10 +15,10 @@
 
 while(<>){$doc .= $_}
 
-$* = 1; # Match multiple lines
+#$* = 1; # Match multiple lines (depreciated)
 
 # Get rid of <!-- ... --> comments
-$doc =~ s|<\!\-\-.*?\-\->||sg;
+$doc =~ s|<\!\-\-.*?\-\->||sgm;
 
 # body flags
 $bodyflags = "";
@@ -27,11 +27,11 @@
 if($doc =~ m|<head>(.*?)</head>|is) {
     $header = $1;
     # Get rid of the TH; HTML doesn't use it
-    $header =~ s|<th>.*?</th>||igs;
+    $header =~ s|<th>.*?</th>||igsm;
     # Get rid of DTWIDTH too
-    $header =~ s|<dtwidth>.*?</dtwidth>||igs;
+    $header =~ s|<dtwidth>.*?</dtwidth>||igsm;
     # Parse the body flags
-    if($header =~ s|<bodyflags>(.*?)</bodyflags>||igs) {
+    if($header =~ s|<bodyflags>(.*?)</bodyflags>||igsm) {
         $bodyflags = $1;
 	}
     }
@@ -69,7 +69,7 @@
 # Enough of header processing; let's get to the body of the document
 
 # Grab the body
-if($doc =~ m|<body>(.*?)</body>|is) {
+if($doc =~ m|<body>(.*?)</body>|ism) {
     $body = $1;
     }
 else {
@@ -91,17 +91,17 @@
    my($hack,$filename);
 
    # The INCLUDE and HINCLUDE tags
-   while($body =~ m|\<h?include\s+\"([^"]+)\"\s*\>|is) {
+   while($body =~ m|\<h?include\s+\"([^"]+)\"\s*\>|ism) {
        $filename = $1;
        open(FILE,"< $filename") || die "Can not find file $filename\n";
        $hack = "";
        while(<FILE>) {$hack .= $_}
        close(FILE);
        #$hack = process_body($hack);
-       $body =~ s|\<h?include\s+\"([^"]+)\"\s*\>|$hack|is;
+       $body =~ s|\<h?include\s+\"([^"]+)\"\s*\>|$hack|ism;
        }
 
-   $body =~ s|\<\/?hibit[^>]*\>||ig;
+   $body =~ s|\<\/?hibit[^>]*\>||igm;
 
    return $body;
 
--- deadwood-2.9.07/tools/ej/ej2txt	2009-05-10 11:15:43.000000000 -0700
+++ deadwood-3.0.01/tools/ej/ej2txt	2010-09-21 10:51:59.000000000 -0700
@@ -4,6 +4,10 @@
 # text document
 
 TMP=$HOME/tmp
+LINKS=$( which elinks )
+if [ -z "$LINKS" ] ; then
+	LINKS=links
+fi
 
 if which ej2html > /dev/null 2>&1 ; then
 	CMD=ej2html
@@ -21,7 +25,7 @@
 fi
 
 $CMD $1 > $TMP/$$.html
-links -dump-width 70 -dump -no-references $TMP/$$.html | awk '
+$LINKS -dump-width 70 -dump -no-references $TMP/$$.html | awk '
 	{gsub(/\[[0-9]+\]/,"");print}' 
 rm $TMP/$$.html
 
