--- deadwood-2.9.07/doc/FAQ.ej	2010-09-08 21:54:51.000000000 -0700
+++ deadwood-3.0.01/doc/FAQ.ej	2010-09-20 09:02:02.000000000 -0700
@@ -90,4 +90,44 @@
 drops very few (if any) DNS packets sent to it.  Namebench is available at
 available at http://code.google.com/p/namebench/
 
+<h2>Can Deadwood blacklist by domain?</h2>
+
+Yes.
+<p>
+
+To blacklist a domain, add a line like this to the dwood3rc file:
+
+<pre>
+upstream_servers["example.com."] = "192.168.255.255"
+</pre>
+
+Replace "example.com." with the domain to be blacklisted, and replace 
+"192.168.255.255" with an IP that is either:
+
+<ul>
+
+<li>Guaranteed to be unreachable from the server running Deadwood.  In this
+    case, any attempt to reach a blacklisted domain will result in Deadwood
+    timing out and eventually returning a "SERVER FAIL" error message.
+
+<li>Running a DNS server that always returns the same IP to a given DNS
+    query.  This can be set up by using "csv2_default_zonefile" in MaraDNS,
+    running "microdns" (included in MaraDNS' source tree), or even by running 
+    "nanodns".
+
+</ul>
+
+Deadwood uses a hash to store these blacklisted domains, and should be able
+to store thousands of such domains without significant slowdown.
+<p>
+
+If it is more convenient to store the domains in separate files, this can be
+done using Deadwood's "execfile" mechanism.
+
+<h2>Does Deadwood have DNSSEC support?</h2>
+
+No.  I have nothing against DNSSEC per se, but I plain simply am not in a
+position to take the time and effort to implement DNSSEC without being
+compensated for my work.
+
 </BODY>
