--- deadwood-3.0.03/src/DwMararc.c	2011-07-22 15:41:30.000000000 -0500
+++ deadwood-3.0.04/src/DwMararc.c	2011-09-10 13:41:10.389630157 -0500
@@ -567,6 +567,8 @@
         if(key_d[num] == 0 && todo == 6) { /* Initialized with {} */
                 key_d[num] = dwd_init();
                 return;
+	} else if(key_d[num] != 0 && todo == 6) { /* Only initialize once */	
+		dwm_fatal("Dictionary variable already initialized");
         } else if(key_d[num] == 0) {
                 dwm_fatal("Uninitialized dictionary variable");
         }
