--- deadwood-3.0.03/src/DwUdpSocket.c	2011-07-22 15:41:24.000000000 -0500
+++ deadwood-3.0.04/src/DwUdpSocket.c	2011-09-10 10:52:12.407405705 -0500
@@ -198,6 +198,12 @@
         /* Get IP of remote server and open a socket to them */
         for(counter = 0; counter < 7; counter++) { /* Don't stop on bad IP */
                 addr = get_upstream_ip(query,n);
+		/* Make sure RD is appropriately set or clear */
+		if(rem[n].is_upstream == 1) {
+			packet[2] |= 0x01; /* Set RD */
+		} else {
+			packet[2] &= 0xfe; /* Clear RD */
+		}
                 if(addr.len > 0) {
                         break;
                 }
