--- deadwood-S-20110910-2/sqa/sqa_tcp/do.test.orig	2011-09-10 15:25:40.470629912 -0500
+++ deadwood-S-20110910-2/sqa/sqa_tcp/do.test	2011-09-10 15:25:59.410493205 -0500
@@ -25,7 +25,7 @@
 dig +tcp @127.0.0.1 www.example.com | \
 	grep -v DiG | grep -v WHEN | grep -v time | grep -v HEADER | \
 	grep -v 'WARNING: recursion requested but not available' | \
-	grep -v NS
+	grep -v 'global options' | grep -v NS
 killall maradns > /dev/null 2>&1
 killall zoneserver > /dev/null 2>&1
 killall Deadwood
--- deadwood-S-20110910-2/sqa/sqa_tcp/output.success.orig	2011-09-10 15:26:09.162629716 -0500
+++ deadwood-S-20110910-2/sqa/sqa_tcp/output.success	2011-09-10 15:26:18.005649621 -0500
@@ -1,6 +1,5 @@
 
 ; (1 server found)
-;; global options: +cmd
 ;; Got answer:
 
 ;; QUESTION SECTION:
--- deadwood-S-20110910-2/sqa/do.tests.orig	2011-09-10 16:22:39.019627935 -0500
+++ deadwood-S-20110910-2/sqa/do.tests	2011-09-10 16:23:16.413445372 -0500
@@ -55,6 +55,8 @@
 
 # These tests are stable and don't fail
 TESTS=$TESTS"basic_test basic_tcp_test "
+TESTS=$TESTS"sqa_server_fail "
+TESTS=$TESTS"sqa_root_upstream "
 TESTS=$TESTS"dwood2rc_s_bind_address "
 TESTS=$TESTS"dwood2rc_n_dns_port,3 "
 TESTS=$TESTS"dwood2rc_n_filter_rfc1918 "
--- deadwood-S-20110910-3/sqa/dwood2rc_n_max_ar_chain/do.test.orig	2011-09-10 18:00:50.806493232 -0500
+++ deadwood-S-20110910-3/sqa/dwood2rc_n_max_ar_chain/do.test	2011-09-10 18:21:27.407384792 -0500
@@ -37,6 +37,7 @@
 recursive_acl="127.0.0.1/16"
 max_ar_chain = ${MAX_AR_CHAIN}
 filter_rfc1918 = 0
+ttl_age = 0
 EOF
 
 ../../src/Deadwood -f dwood2rc | grep -v version | sed 's/[(].*[)]//' &
