--- deadwood-3.0.04/doc/internals/MARARC.parser.orig	2011-09-11 23:23:49.486630132 -0500
+++ deadwood-3.0.04/doc/internals/MARARC.parser	2011-09-11 23:26:20.418401313 -0500
@@ -52,7 +52,7 @@
 in the quoted state machine definition; #: So we can potentially
 add comments to state machine definitions)
 
-We also have seven actions:
+We also have eight actions:
 
 1: Add the character we are looking at to variable 1
 2: Add the character we are looking at to variable 2
@@ -60,6 +60,8 @@
 4: Add the character we are looking at to variable 4
 5: Add the character we are looking at to variable 5
 6: Add the character we are looking at to variable 6
+7: Add the character we are looking at to variable 7
+8: Return a fatal error stating that leading whitespace is not allowed
 ;: End the processing of the current line successfully
 
 The variables are:
@@ -119,7 +121,7 @@
 At left paren:			     y: Qz
 In filename for execfile:	     z: I7z Qxa
 Quote after execfile filename:	     xa: )k
-After whitespace in line:            xb: Hb Wxb Rxp T;
+After whitespace in line:            xb: Hb Wxb Y8xb Rxp T;
 
 Once a line is processed, we then look at the value of variable 1 (the 
 mararc parameter):
--- deadwood-3.0.04/src/DwMararc.h.orig	2011-09-11 23:23:32.135629818 -0500
+++ deadwood-3.0.04/src/DwMararc.h	2011-09-11 23:26:26.888789121 -0500
@@ -213,7 +213,7 @@
                     "y Qz\n" \
                     "z I7z Qxa\n" \
                     "xa )k\n" \
-                    "xb Hb Wxb Rxp T;\n"
+                    "xb Hb Wxb Y8xb Rxp T;\n"
 
 /* A tokenized single state in the finite state machine */
 typedef struct {
--- deadwood-3.0.04/src/DwMararc.c.orig	2011-09-11 23:23:23.930630077 -0500
+++ deadwood-3.0.04/src/DwMararc.c	2011-09-11 23:30:27.598379723 -0500
@@ -710,6 +710,9 @@
                 if(action == 10) { /* Terminate */
                         ret = 0; /* Line parsed */
                         break; /* Now we need to set the keys */
+		} else if(action == 8) {
+			ret = -4; /* Fatal: No leading whitespace */
+			goto catch_dwm_parse_line;
                 } else if(action > 0 && action < 10) {
                         dwm_do_action(ch,action,actions);
                 }
@@ -754,7 +757,9 @@
                 a = dwm_parse_line(look);
                 if(a == -3) {
                         dwm_fatal("incomplete last line");
-                }
+                } else if(a == -4) {
+			dwm_fatal("leading whitespace not allowed");
+		}
                 if(a != 0 && a != 1) {
                         dwm_fatal("deadwoodrc parse error");
                 }
