--- deadwood-3.0.03/doc/CHANGELOG	2011-07-22 15:50:34.000000000 -0500
+++ deadwood-3.0.04/doc/CHANGELOG	2011-11-01 09:59:00.422686719 -0600
@@ -1,3 +1,32 @@
+3.0.04 (20111101):
+
+* Bugfix: RD should now be correctly set (or cleared) when sending a query
+          upstream.
+
+* Parser update: upstream_servers and root_servers nodes can not be 
+  newly defined more than once.
+
+* Parser update: dwood3rc files can no longer have leading whitespace before
+  definitions.
+
+* Parser update: Dictionary variables can no longer be initialized twice
+
+* SQA update: 2 new SQA tests added (sqa_root_upstream: Make sure we can have
+  both set in the same dwood3rc file; sqa_server_fail: Make sure 
+  upstream_servers works when one of the two upstream servers is returning
+  a SERVER FAIL); SQA tests updated to pass in both CentOS 5 and Scientific
+  Linux 6.
+
+* It is no longer possible to have a bind_address that is not in one's
+  recursive_acl
+
+* filter_rfc1918 parameter added to example dwood3rc file
+
+* reject_ptr parameter added; both reject_aaaa and reject_ptr now immediately
+  return a "not there" (instead of waiting for a reply upstream)
+
+* Deadwood-HOWTO and UPDATE-guide added to documentation
+
 3.0.03 (20110722):
 
 * Bugfix: answers.yahoo.com problem fixed (we now extract one useful 
--- deadwood-3.0.03/doc/Changelog.summary	2010-07-22 11:33:02.000000000 -0500
+++ deadwood-3.0.04/doc/Changelog.summary	2011-11-01 10:01:02.864698210 -0600
@@ -32,3 +32,5 @@
 
 * reject_aaaa ; feature-complete testing release (2.9.01)
 
+* stable release of fully recursive Deadwood (3.0.01)
+
