diff -ur maradns-1.4.07/libs/MaraHash.c maradns-1.4.08/libs/MaraHash.c
--- maradns-1.4.07/libs/MaraHash.c	2011-11-11 16:58:34.000000000 -0600
+++ maradns-1.4.08/libs/MaraHash.c	2011-12-29 11:37:10.311564695 -0600
@@ -1,4 +1,4 @@
-/* Copyright (c) 2006 Sam Trenholme
+/* Copyright (c) 2006, 2011 Sam Trenholme
  *
  * TERMS
  *
@@ -32,6 +32,7 @@
 #include "JsStr.h"
 #endif
 #include "MaraHash.h"
+#include <stdio.h>
 
 /* Masks to limit the size of the hash */
 /* These are powers of two, minus one */
@@ -41,6 +42,8 @@
                           16777215, 33554431, 67108863, 134217727,
                           268435455, 536870911, 1073741823 };
 
+mhash_offset mhash_secret_add_constant = 7;
+
 /* Create a new, blank mhash object
    input: none
    output: pointer to the object in quesiton on success, NULL (0)
@@ -100,7 +103,7 @@
     /* Simple enough hash */
     while(point < max) {
         ret += (mhash_offset)(*point << shift);
-        shift += 7;
+        shift += mhash_secret_add_constant;
         shift %= hash_bits;
         point++;
         }
@@ -684,3 +687,21 @@
     return tuple->tuple_list[element];
     }
 
+/* Read three bytes from a filename and use that as a secret add constant */
+int mhash_set_add_constant(char *filename) {
+	FILE *read = 0;
+
+	read = fopen(filename,"rb");
+	if(read == NULL) {
+		return -1;
+	}
+	
+	mhash_secret_add_constant ^= getc(read);
+	mhash_secret_add_constant <<= 8;
+	mhash_secret_add_constant ^= getc(read);
+	mhash_secret_add_constant <<= 8;
+	mhash_secret_add_constant ^= getc(read);
+	fclose(read);
+	return 1;
+}
+
diff -ur maradns-1.4.07/server/MaraDNS.c maradns-1.4.08/server/MaraDNS.c
--- maradns-1.4.07/server/MaraDNS.c	2011-11-11 16:58:33.000000000 -0600
+++ maradns-1.4.08/server/MaraDNS.c	2011-12-29 11:41:54.315941513 -0600
@@ -1,4 +1,4 @@
-/* Copyright (c) 2002-2010 Sam Trenholme
+/* Copyright (c) 2002-2011 Sam Trenholme
  *
  * TERMS
  *
@@ -3952,6 +3952,24 @@
     int recurse_number_ports = 4096;
 #endif
 
+    /* First order of business: Initialize the hash */
+    if(mhash_set_add_constant(
+#ifdef MINGW32
+	"secret.txt"
+#else
+	"/dev/urandom"
+#endif
+		) != 1) {
+	printf(
+#ifdef MINGW32
+	"Fatal error opening secret.txt"
+#else
+	"Fatal error opening /dev/urandom"
+#endif
+			);
+			return 32;
+	}
+    
     memset(&client,0,sizeof(client)); /* Initialize ya variables */
     clin = (struct sockaddr_in *)&client;
 #ifdef AUTHONLY
