--- maradns-1.2.12.03/doc/en/source/changelog.embed	2006-10-08 19:32:42.000000000 -0400
+++ maradns-1.2.12.04/doc/en/source/changelog.embed	2006-11-24 06:38:53.000000000 -0500
@@ -1,5 +1,27 @@
 <H1>MaraDNS changelog</H1>
 
+<p>maradns-1.2.12.04:
+<p>
+This is a stable release of MaraDNS.
+<blockquote>
+<ul>
+<li>Bugfix: <tt>make debug</tt> works again.  Note that <tt>make debug</tt>
+    <b>really</b> slows MaraDNS down.
+<li><tt>ej2man</tt> now works with both Perl 5.8.0 and 5.8.8.  I would like
+    to thank the people on comp.lang.perl.misc for their help.
+<li><tt>ej2txt</tt> now works with different versions of <tt>links</tt>.
+<li>Fixed bug with ANY and case sensitivity.  I would like to thank Kyler 
+    Laird for pointing out this issue.
+<li>Added new mararc variable, <tt>csv2_tilde_handling</tt> the currently
+    does nothing but require, if set, having the value of 0.  This is here
+    so a MaraDNS 1.2.12 mararc is compatible with some future plans I have.
+<li>Added new compiler define <tt>UPSTREAM_PORT</tt> in <tt>MaraDns.h</tt>
+    that allows the port MaraDNS uses to connect to other nameserver to be
+    more easily changed.
+</ul>
+(2006.11.24)
+</blockquote>
+
 <p>maradns-1.2.12.03:
 <p>
 This is a stable release of MaraDNS.
