--- deadwood-2.9.02/src/DwSys.c	2010-07-29 10:39:10.000000000 -0700
+++ deadwood-2.9.03/src/DwSys.c	2010-08-05 02:50:13.000000000 -0700
@@ -617,10 +617,16 @@
            /* a[8], a[9]: NSCOUNT */
            a[8] != 0 || a[9] != 0 ||
            /* a[10], a[11]: ARCOUNT */
-           a[10] != 0 || a[11] != 0))
+           a[10] != 0))
         {
                 return -1;
         }
+	if(roy_arends_check == 2 && a[11] == 1) {
+		return -2; /* Send NOTIMPL; OPT/EDNS not supported */
+	}
+	if(roy_arends_check == 2 && a[11] != 0) {
+		return -1; /* Drop packet */
+	}
         /* A DNS header is not considered kosher if it doesn't
          * have an question, nor any other RRs ; we look
          * at QDCOUNT, ANCOUNT, NSCOUNT, and ARCOUNT */
--- deadwood-2.9.03/src/DwUdpSocket.c.orig	2010-08-05 02:50:13.000000000 -0700
+++ deadwood-2.9.03/src/DwUdpSocket.c	2010-08-05 03:04:33.000000000 -0700
@@ -617,6 +617,10 @@
         len = recvfrom(sock,(void *)packet,520,0,(struct sockaddr *)&client,
                        &c_len);
 
+	if(len < 12) {
+                goto catch_get_local_udp_packet;
+        }
+
         from_port = get_from_ip_port(&from_ip,&client);
 
         if(check_ip_acl(&from_ip) != 1) { /* Drop unauthorized packets */
@@ -624,6 +628,13 @@
         }
 
         local_id = get_dns_qid(packet, len, 2);
+	if(local_id == -2) { /* Immediate NOTIMPL for EDNS requests w/ OPT */
+		packet[2] = 0x81; /* QR = 1; Op = 0; AA = 0; TC = 0; RD = 1 */
+		packet[3] = 0x84; /* RA = 1; Z = 0; RCODE = "notimpl" (4) */
+		sendto(sock,(void *)packet,len,0,(struct sockaddr *)&client,
+				c_len);
+		goto catch_get_local_udp_packet;
+	}	
         if(local_id < 0 || len < 13) { /* Invalid remote packet */
                 goto catch_get_local_udp_packet;
         }
