--- deadwood-2.9.02/src/DwSocket.c	2010-07-29 10:39:10.000000000 -0700
+++ deadwood-2.9.03/src/DwSocket.c	2010-08-02 11:06:26.000000000 -0700
@@ -293,6 +293,8 @@
 
         q = strchr(c,'/');
         if(q == 0 || *q != '/') {
+		pop_ip_core(list,(ip_addr_T *)ip_mask,c);
+		make_netmask(ip_mask->len * 8,ip_mask->mask,ip_mask->len);
                 goto catch_pop_ip_mask;
         }
         *q = 0; /* pop_ip_core can't take an IP ending with a slash */
