--- build/rpm.mararc.orig	2013-11-27 10:02:01.000000000 -0500
+++ build/rpm.mararc	2013-11-27 10:02:38.000000000 -0500
@@ -1,6 +1,6 @@
 # Default mararc file for rpm build of MaraDNS.
 # This only shows a subset of MaraDNS' features needed to be an
-# authoritative and recursive name server.  Look at 
+# authoritative name server.  Look at 
 # detailed/example_full_mararc for an example showing most of 
 # the features that MaraDNS has.
 
@@ -16,13 +16,6 @@
 # The directory with all of the zone files
 chroot_dir = "/etc/maradns"
 
-# The following line enables recursion on just the loopback interface
-recursive_acl = "127.0.0.1/8"
-
-# If you're on a slow network, you may need to increase this value for
-# MaraDNS to resolve names
-timeout_seconds = 2
-
 # The various zones a MaraDNS server hosts; these zones are just examples
 
 # When running in authoritative mode, we must initialize the csv2 hash, 
