--- deadwood-3.2.04/doc/CHANGELOG	2013-12-20 07:48:10.000000000 -0800
+++ deadwood-3.2.05/doc/CHANGELOG	2014-02-11 09:37:18.654557100 -0800
@@ -1,3 +1,20 @@
+3.2.05 (2014-02-11):
+
+- Security update: Certain malformed DNS packets could crash Deadwood 
+  (especially in Windows)
+
+- Snapshot releases now in YYYY-MM-DD-N form
+
+- Fixed cppcheck detected warnings and errors
+
+- Default file name for entropy in Windows is now "secret.txt"
+
+- dw_substr() bounds check updated
+
+- More validation done with DNS packets sent to Deadwood
+
+- Dashes can now be the first character of a black listed domain
+
 3.2.04 (20131220):
 
 - Security patch as described at http://samiam.org/blog/20131202.html
