diff -ur deadwood-3.0.02/sqa/sqa_tcp/do.test deadwood-3.0.03/sqa/sqa_tcp/do.test
--- deadwood-3.0.02/sqa/sqa_tcp/do.test	2010-01-19 10:32:48.000000000 -0600
+++ deadwood-3.0.03/sqa/sqa_tcp/do.test	2011-06-24 10:52:07.296507143 -0500
@@ -11,7 +11,7 @@
 # real www.example.com auth server; change this if www.example.com's DNS
 # moves
 upstream_servers = {}
-upstream_servers["."]="192.0.34.43"
+upstream_servers["."]="199.43.132.53"
 recursive_acl="127.0.0.1/16"
 tcp_listen = 1
 timeout_seconds_tcp = 10
@@ -23,7 +23,9 @@
 
 sleep 3
 dig +tcp @127.0.0.1 www.example.com | \
-	grep -v DiG | grep -v WHEN | grep -v time | grep -v HEADER
+	grep -v DiG | grep -v WHEN | grep -v time | grep -v HEADER | \
+	grep -v 'WARNING: recursion requested but not available' | \
+	grep -v NS
 killall maradns > /dev/null 2>&1
 killall zoneserver > /dev/null 2>&1
 killall Deadwood
diff -ur deadwood-3.0.02/sqa/sqa_tcp/output.success deadwood-3.0.03/sqa/sqa_tcp/output.success
--- deadwood-3.0.02/sqa/sqa_tcp/output.success	2009-11-11 07:40:32.000000000 -0600
+++ deadwood-3.0.03/sqa/sqa_tcp/output.success	2011-06-24 10:52:38.179522788 -0500
@@ -1,19 +1,15 @@
 
 ; (1 server found)
-;; global options:  printcmd
+;; global options: +cmd
 ;; Got answer:
-;; flags: qr aa rd; QUERY: 1, ANSWER: 1, AUTHORITY: 2, ADDITIONAL: 0
 
 ;; QUESTION SECTION:
 ;www.example.com.		IN	A
 
-;; ANSWER SECTION:
-www.example.com.	172800	IN	A	192.0.32.10
+www.example.com.	172800	IN	A	192.0.43.10
 
 ;; AUTHORITY SECTION:
-example.com.		172800	IN	NS	a.iana-servers.net.
-example.com.		172800	IN	NS	b.iana-servers.net.
 
 ;; SERVER: 127.0.0.1#53(127.0.0.1)
-;; MSG SIZE  rcvd: 112
+;; MSG SIZE  rcvd: 97
 
