--- maradns-1.3.07.03/server/MaraDNS.c.orig	2007-08-16 08:01:27.000000000 -0500
+++ maradns-1.3.07.03/server/MaraDNS.c	2007-08-16 08:10:58.000000000 -0500
@@ -3018,7 +3018,9 @@
 
     nxstore = NULL;
 
-    do {
+    if(has_recursive_authority != 1 || desires_recursion != 1 
+       || recurse_delegation != 1) {
+      do {
         spot_data = mhash_get(bighash,lookfor);
         point = spot_data.value;
         /* We stop going up the tree if we have found an authoritative NS
@@ -3046,6 +3048,7 @@
             return JS_SUCCESS;
             }
         } while(bobbit_label(lookfor) > 0);
+    }
 
 skip_delegation_ns:
 
