--- maradns-1.2.07.2/build/maradns-1.2.07.2.spec	2006-04-05 02:11:48.000000000 -0700
+++ maradns-1.2.07.3/build/maradns-1.2.07.3.spec	2006-05-15 16:10:14.000000000 -0700
@@ -1,12 +1,12 @@
 Summary: An authoritative and recursive DNS server made with security in mind
 Name: maradns
-Version: 1.2.07.2
+Version: 1.2.07.3
 Release: 1
 License: BSD (Two-clause)
 Group: Networking/Daemons
-Source: http://www.maradns.org/download/1.2/1.2.07.2/maradns-1.2.07.2.tar.bz2
-Source1: maradns-1.2.07.2.tar.bz2.sha1.asc
-Source2: maradns-1.2.07.2.tar.bz2.rmd.asc
+Source: http://www.maradns.org/download/1.2/1.2.07.3/maradns-1.2.07.3.tar.bz2
+Source1: maradns-1.2.07.3.tar.bz2.sha1.asc
+Source2: maradns-1.2.07.3.tar.bz2.rmd.asc
 Patch0: maradns-1.1.59-rpm.patch
 BuildRoot: /var/tmp/%{name}-buildroot
 
@@ -49,7 +49,7 @@
 
 %files
 %defattr(-,root,root)
-%doc /usr/doc/maradns-1.2.07.2/*
+%doc /usr/doc/maradns-1.2.07.3/*
 
 /usr/sbin/maradns
 /usr/sbin/zoneserver
@@ -69,6 +69,7 @@
 /usr/man/man5/mararc.5*
 /etc/rc.d/rc3.d/S60maradns
 /etc/rc.d/rc5.d/S60maradns
+/etc/maradns/logger
 %config /etc/mararc
 %config /etc/maradns/db.example.net
 %config /etc/rc.d/init.d/maradns
@@ -87,6 +88,9 @@
 echo MaraDNS should have been stopped
 
 %changelog
+* Wed Apr 19 2006 Sam Trenholme <sam+i55e6bt@chaosring.org>
+- MaraDNS rpm package updated for version 1.2.07.3
+
 * Tue Mar 28 2006 Sam Trenholme <sam+i55e6bt@chaosring.org>
 - MaraDNS rpm package updated for version 1.2.07.2
 
