--- maradns-1.3.07.03/doc/en/source/changelog.embed	2007-08-16 08:35:43.000000000 -0500
+++ maradns-1.3.07.04/doc/en/source/changelog.embed	2007-08-29 11:23:45.000000000 -0500
@@ -1,5 +1,46 @@
 <H1>MaraDNS changelog</H1>
 
+<p>maradns-1.3.07.04:</p>
+This is a beta testing release of MaraDNS.
+<blockquote>
+<ul>
+<li>Bugfix: Cause of improper rotation of resource records found and
+    fixed.  I would like to thank Michael Krieger for his bug report.
+<li>Cleanup: MinGW32 compile-time warnings removed
+<li>Documentation: All SQA regressions now have README files; zoneserver
+    man page now has example mararc file
+<li>Patch contributed by Joerg Sonnenberger making things easier to compile
+    in DragonflyBSD
+<li>Install.sh directory for documentation fixed.  I would like to
+    thank Brandon Holbrook for his bug report.
+<li>SQA meta-script that runs and verifies all SQA regressions added
+<li>SQA tests added: Spurious "bad query" messages, "percent" parse bug
+(doesn't seem to ever have been a real bug), "noat" parse bug, Star handling
+when <tt>bind_star_handling</tt> has a value of 1, improper RR rotation bug,
+and making sure the example mararc file in the zoneserver man page is correct
+</ul>
+(2007.08.29)
+</blockquote>
+
+<p>maradns-1.2.12.07:</p>
+This is a stable bugfix-only release of MaraDNS.  All of these bug fixes
+are backports of bugs fixed in the 1.3 branch.
+<blockquote>
+<ul>
+<li>Bugfix: If bind_star_handling was set to 2, MaraDNS would leak
+    memory when the existance of another RR stopped resolution using a
+    star record.
+<li>Bugfix: bind_star_handling, when set to 2, now appears to do the
+    right thing.
+<li>Bugfix: Non-critical double-free() removed.
+<li>Bugfix: askmara no longer goes in to an infinite loop when it
+    receives an invalid TXT record.
+<li>Bugfix: csv2 parser now makes sure that TXT chunks are no longer
+    than 255 characters in all cases.
+</ul>
+(2007.08.16)
+</blockquote>
+
 <p>maradns-1.3.07.03:</p>
 This is a beta testing release of MaraDNS.
 <blockquote>
