--- maradns-1.4.03/doc/en/webpage/download.embed	2010-02-02 09:12:44.000000000 -0800
+++ maradns-1.4.04/doc/en/webpage/download.embed	2010-07-30 23:19:08.000000000 -0700
@@ -3,49 +3,47 @@
 <h1>&iexcl;Download MaraDNS!</h1>
 
 <h2>Stable release</h2>
-<p>This is the current stable release of MaraDNS, MaraDNS 1.4.03,
-released February 2, 2010.
-<!--
+<p>This is the current stable release of MaraDNS, MaraDNS 1.4.04,
+released July 31, 2010.
 <p>This release is available both at <A 
 href="http://sourceforge.net/project/showfiles.php?group_id=24033">sourceforge.net</A>
 and on this page.
--->
 <p>
 
 <table><tr><td>
 <A 
-href="download/1.4/1.4.03/maradns-1.4.03.tar.bz2"><!--
-                       -->maradns-1.4.03.tar.bz2</A>
+href="download/1.4/1.4.04/maradns-1.4.04.tar.bz2"><!--
+                       -->maradns-1.4.04.tar.bz2</A>
 </td><td>
 <A 
-href="download/1.4/1.4.03/maradns-1.4.03.tar.bz2.asc">sig</A>
+href="download/1.4/1.4.04/maradns-1.4.04.tar.bz2.asc">sig</A>
 </td><td>
 <A href="#bz2">info</A> 
 </td></tr><tr><td>
 <A 
-href="download/1.4/1.4.03/maradns-1.4.03.tar.gz"><!--
-                       -->maradns-1.4.03.tar.gz</A>
+href="download/1.4/1.4.04/maradns-1.4.04.tar.gz"><!--
+                       -->maradns-1.4.04.tar.gz</A>
 </td><td>
 <A 
-href="download/1.4/1.4.03/maradns-1.4.03.tar.gz.asc">sig</A>
+href="download/1.4/1.4.04/maradns-1.4.04.tar.gz.asc">sig</A>
 </td><td>
 <A href="#gz">info</A>
 </td></tr><tr><td>
 <A 
-href="download/1.4/1.4.03/maradns-1-4-03-win32.zip"><!--
-                       -->maradns-1-4-03-win32.zip</A>
+href="download/1.4/1.4.04/maradns-1-4-04-win32.zip"><!--
+                       -->maradns-1-4-04-win32.zip</A>
 </td><td>
 <A 
-href="download/1.4/1.4.03/maradns-1-4-03-win32.zip.asc">sig</A>
+href="download/1.4/1.4.04/maradns-1-4-04-win32.zip.asc">sig</A>
 </td><td>
 <A href="#zip">info</A>
 </td></tr><tr><td>
 <A 
-href="download/1.4/1.4.03/maradns-1-4-03-win32.7z"><!--
-                       -->maradns-1-4-03-win32.7z</A>
+href="download/1.4/1.4.04/maradns-1-4-04-win32.7z"><!--
+                       -->maradns-1-4-04-win32.7z</A>
 </td><td>
 <A 
-href="download/1.4/1.4.03/maradns-1-4-03-win32.7z.asc">sig</A>
+href="download/1.4/1.4.04/maradns-1-4-04-win32.7z.asc">sig</A>
 </td><td>
 <A href="#7z">info</A>
 </td></tr></table>
@@ -54,28 +52,28 @@
 
 <h2>Deadwood</h2>
 
-Deadwood is the code that will eventually become MaraDNS' next recursive
-resolver; right now it is a stable and functioning non-recursive cache.
+Deadwood is the code that will soon become MaraDNS' next recursive
+resolver; right now it is a recursive cache undergoing beta testing.
 Deadwood, unlike MaraDNS, does not need threads to resolve queries, and
-has features MaraDNS' recursive cache doesn't have, such as "resurrections"
-and the ability to read and write the cache to disk.
+has features MaraDNS 1.x's recursive code doesn't have, such as 
+"resurrections" and the ability to read and write the cache to disk.
 
 <table><tr><td>
 <A 
-href="deadwood/stable/deadwood-2.4.10.tar.bz2"><!--
-                   -->deadwood-2.4.10.tar.bz2</A>
+href="deadwood/testing/deadwood-2.9.02.tar.bz2"><!--
+                    -->deadwood-2.9.02.tar.bz2</A>
 </td><td>
 <A 
-href="deadwood/stable/deadwood-2.4.10.tar.bz2.asc">sig</A>
+href="deadwood/testing/deadwood-2.9.02.tar.bz2.asc">sig</A>
 </td><td>
 <A href="#bz2">info</A> 
 </td></tr><tr><td>
 <A 
-href="deadwood/stable/Deadwood-2-4-10-win32.zip"><!--
-                   -->Deadwood-2-4-10-win32.zip</A>
+href="deadwood/testing/Deadwood-2-9-02-win32.zip"><!--
+                    -->Deadwood-2-9-02-win32.zip</A>
 </td><td>
 <A 
-href="deadwood/stable/Deadwood-2-4-10-win32.zip.asc">sig</A>
+href="deadwood/testing/Deadwood-2-9-02-win32.zip.asc">sig</A>
 </td><td>
 <A href="#zip">info</A>
 </td></tr></table>
@@ -105,7 +103,7 @@
                           -->maradns-1.3.07.10.tar.bz2</A>
 </td><td>
 <A 
-href="download/1.3/1.3.07.10/maradns-1.3.07.10.tar.bz2.sha.asc">sig</A>
+href="download/1.3/1.3.07.10/maradns-1.3.07.10.tar.bz2.asc">sig</A>
 </td><td>
 <A href="#bz2">info</A> 
 </td></tr>
@@ -115,7 +113,7 @@
                           -->maradns-1.3.07.10.tar.gz</A>
 </td><td>
 <A 
-href="download/1.3/1.3.07.10/maradns-1.3.07.10.tar.gz.sha.asc">sig</A>
+href="download/1.3/1.3.07.10/maradns-1.3.07.10.tar.gz.asc">sig</A>
 </td><td>
 <A href="#gz">info</A>
 </td></tr></table>
