--- deadwood-2.9.07/src/DwUdpSocket.c.orig	2010-09-07 22:18:45.000000000 -0700
+++ deadwood-2.9.07/src/DwUdpSocket.c	2010-09-07 22:20:02.000000000 -0700
@@ -833,7 +834,7 @@
                 goto catch_verify_dns_packet;
         }
 
-        /* Should we make this case-insensitive? Probably not. */
+	dwc_lower_case(question); /* Case-insensitive comparison */
         if(dw_issame(question,rem[b].query) != 1) {
                 goto catch_verify_dns_packet;
         }
