--- deadwood-2.9.07/src/DwUdpSocket.c.orig	2010-09-06 18:16:09.000000000 -0700
+++ deadwood-2.9.07/src/DwUdpSocket.c	2010-09-06 18:17:19.000000000 -0700
@@ -517,7 +517,7 @@
         }
 #endif
 
-        dw_log_dwstr("Sending reply for query ",query,100);
+        dw_log_dwstr("Looking in cache for query ",query,100);
         type = dw_fetch_u16(query,-1);
 	if(type == RR_MX && key_n[DWM_N_reject_mx] != 0) {
 		dw_log_ip(
@@ -526,6 +526,10 @@
 	}	
         dwc_process(cache,query,3); /* RR rotation, TTL aging, etc. */
         value = dwh_get(cache,query,resurrect,1);
+	if(value == 0) {
+		dw_log_dwstr("Nothing found for ",query,100);
+		goto catch_get_reply_from_cache;
+	}
         cache_type = dw_fetch_u8(value,-1);
         if(cache_type != TYPE_TRUNCATED &&
                         cache_type != TYPE_TRUNCATED_NXDOMAIN) {
