--- maradns-1.3.07.04/doc/en/source/faq.embed	2007-03-22 01:31:27.000000000 -0600
+++ maradns-1.3.07.05/doc/en/source/faq.embed	2007-09-19 18:49:18.000000000 -0500
@@ -77,21 +77,26 @@
     <p>
 
 32. <A href="#synthns">I am transferring a zone from another
-    server, but the NS records are these strange "synth-ip" records.</A>
+    server, but the NS records are these strange "synth-ip" records.</A><p>
 
-33. <A href="#roothints">Where is the root.hints file?</A>
+33. <A href="#roothints">Where is the root.hints file?</A><p>
 
 34. <A href="#autoconf">Are there any plans to use autoconf to build 
-    MaraDNS?</A>
+    MaraDNS?</A><p>
 
 35. <A href="#flags">How do I change the compiler or compile-time flags
-    with MaraDNS' build process?</A>
+    with MaraDNS' build process?</A><p>
 
 36. <A href="#makepkg">Will you make a package for the particular Linux 
-    distribution I am using?</A>
+    distribution I am using?</A><p>
 
 37. <A href="#wincrippled">I am using the native Windows port of MaraDNS,
-    and some features are not working.</A>
+    and some features are not working.</A><p>
+
+38. <A href="#startupfail">MaraDNS isn't starting up</A><p>
+
+39. <A href="#releases">You make a lot of releases of MaraDNS; at our
+    ISP/IT department, updating software is non-trivial.</A>
 
 </BLOCKQUOTE>
 <H2>ANSWERS</H2>
@@ -100,8 +105,8 @@
 </A>
 <H2>1. I'm still using version 1.0 of MaraDNS</H2>
 
-MaraDNS 1.0 will continue to be fully supported until December 21, 2007; this
-means that MaraDNS 1.0 questions will still be answered and bug fixes will
+MaraDNS 1.0 will continue to be supported until December 21, 2007; this
+means that MaraDNS 1.0 bug fixes will
 still be applied.  After 2007/12/21, MaraDNS 1.0 will no longer be fully 
 supported; the only updates, at that point, would be bugtraq-worthy critical
 security fixes.  Not even these security updates will be applied after
@@ -147,10 +152,10 @@
 </A>
 <H2>3. What license is MaraDNS released under?</H2>
 
-MaraDNS 1.2 is released with the following two-clause BSD license:
+MaraDNS 1.2 is released with the following two-clause 
 BSD-type license:
 <BLOCKQUOTE>
-Copyright (c) 2002-2005 Sam Trenholme
+Copyright (c) 2002-2007 Sam Trenholme
 <p>
 TERMS
 <p>
@@ -175,7 +180,10 @@
 
 Please contact me; my email address is at
 <A 
-href="http://www.maradns.org/contact.html">http://www.maradns.org/contact.html</A>
+href="http://www.maradns.org/contact.html">http://www.maradns.org/contact.html</A>.  
+Please be sure to include all information requested there, including
+the operating system you are using, the version of MaraDNS you are using,
+your <tt>mararc</tt> configuration file, and all relevant zone files.
 
 <A NAME=chatter>
 </A>
@@ -321,6 +329,9 @@
 a functioning recursive DNS server.  There is a reason why MaraDNS, pdnsd, and
 BIND 9 all use the multi-threaded model.
 
+<p>I am planning on improving MaraDNS' threaded model to not spawn a thread 
+for each and every uncached request.
+
 <A NAME=wishlist>
 </A>
 <H2>13. I feel that XXX feature should be added to MaraDNS</H2>
@@ -338,8 +349,10 @@
 which have been developed as hobbyist projects (Posadis, Pdnsd, and djbdns)
 are no longer being actively worked on by the primary developer.
 <p>
-I plan on someday adding standards-compliant BIND zone file support.
-After that, I may even add real DNS slave support.
+
+My current plans for MaraDNS are visible on the 
+<A href=roadmap.html>roadmap page</A> for MaraDNS.
+
 <p>
 If I see a large
 MaraDNS community and a strong demand for new features from that community,
@@ -434,7 +447,7 @@
 
 A recursive DNS server is a DNS server that is able to contact other DNS
 servers in order to resolve a given domain name label.  This is the kind 
-of DNS server one points to in /etc/resolve.conf
+of DNS server one points to in <tt>/etc/resolve.conf</tt>
 
 <p>
 
@@ -470,10 +483,11 @@
 </A>
 <H2>20. Is MaraDNS portable?</H2>
 
-<p>MaraDNS will only compile on FreeBSD, Mac OS X, Cygwin, Linux, and
-partially on MinGW32 systems.  If you are interested in porting MaraDNS
-to another system, please <A 
-href="http://www.maradns.org/contact.html">let me know</A>.
+<p>MaraDNS is developed on a CentOS 3 and Windows XP dual boot laptop.
+MaraDNS may compile or run on other systems--there are official MaraDNS
+ports for Debian/Ubuntu, Slackware, FreeBSD, and NetBSD.  Note that MaraDNS
+needs a system with a robust threading library, which some systems do not
+have.
 
 <A NAME=cygwin>
 </A>
@@ -921,14 +935,14 @@
 
 <A name=makepkg> </A>
 <h2>Will you make a package for the particular Linux
-    distribution I am using?</A>
+    distribution I am using?</h2>
 
 No.  OK, let me qualify that: I won't do it unless you pay
 me enough money.
 <p>
 
 There are MaraDNS packages for a number of different distributions of
-Linux and other operating systems.   On there MaraDNS site, there is 
+Linux and other operating systems.   On the MaraDNS site, there is 
 a MaraDNS package for CentOS/Red Hat Enterprise Linux available.  There
 is also usually an up-to-date Slackware package available.  In addition,
 there is a Debian package in the Debian packages collection, a FreeBSD
@@ -946,10 +960,31 @@
 <p>
 
 As an aside, some of the MaraDNS packages floating around on the internet
-are out of date.  Please make sure, that if you get a third-party package 
-from the internet, the package is for either MaraDNS 1.0.39, MaraDNS 
-1.2.12.05, or MaraDNS 1.3.03.  Older versions of MaraDNS are not
-supported.
+are out of date (*cough*, Debian, *cough*)<sup><font
+size=-2>1</font></sup>.  Please make sure, that if you get a third-party
+package from the internet, the package is for either MaraDNS 1.0.41,
+MaraDNS 1.2.12.08, or MaraDNS 1.3.07.05.  Older versions of MaraDNS are
+not supported. <p>
+
+Footnote 1: Debian has a somewhat silly policy that, once a package is
+declared "stable", they will basically not update it unless there is a 
+Bugtraq security advisory for the package in question.  This policy is
+a good policy for programs made by pimply-faced 16-year-olds who don't
+know how to manage a release cycle nor a bugfix-only branch, but doesn't
+make sense for MaraDNS.  As I write this, the Debian's "stable" version of
+MaraDNS is 1.2.12.04, which is about a year behind in terms of bugfixes.  
+I, annoyingly enough, get bug reports from Debian users telling me about 
+bugs I have already fixed in the 1.2 branch of MaraDNS.<p>
+
+Now, to be fair to Debian, their policies do allow me to backport bugfixes
+to the 1.2.12.04 release of MaraDNS, and the patches do get reviewed by
+somone else, which minimizes bugfixes introducing new bugs (Yes, I have
+done that), but there are not enough volunteers to review all of the
+bugfixes I have made since 1.2.12.04.  So, Debian users get stuck with
+an old, buggy version of MaraDNS.  The policy would work if there were
+enough volunteers to actually review all of my post-1.2.12.04 bugfixes,
+but the people who created the policy did not take in to account the
+logistics of volunteer work.
 
 <A name=wincrippled> </A>
 <h2>I am using the native Windows port of MaraDNS,
@@ -959,11 +994,11 @@
 Windows port does not have all of the features of the *NIX version of 
 MaraDNS.  In particular, the following features are disabled:
 <ul>
-<li>ipv6
+<li>ipv6 (this is actually a mingw32, not a Windows deficiency)
 <li>The <tt>chroot_dir</tt> mararc variable
 <li>The <tt>maradns_gid</tt> and <tt>maradns_uid</tt> mararc variables
 <li>The <tt>maxprocs</tt> mararc variable
-<li>The <tt>synth_soa_serial</tt> can not have a value of 2
+<li>The <tt>synth_soa_serial</tt> variable can not have a value of 2
 </ul>
 
 If any of the above features are desired, try compiling MaraDNS using
@@ -971,4 +1006,61 @@
 and that while <tt>chroot_dir</tt> works in Cygwin, it does not have
 the security that the *NIX chroot() call has.
 
+<A name=startupfail> </A>
+
+<h2>MaraDNS isn't starting up</h2>
+
+This is usually caused by a syntax error in one's mararc file, or by
+another MaraDNS process already running.  To see what is happening, look
+at your system log (<tt>/var/log/messages</tt> in Centos 3) to see what
+errors MaraDNS reports.  If you do not know how to look at a system
+log, you can also invoke MaraDNS from the command line as root; any errors
+will be visible when starting MaraDNS.
+
+<A name=releases> </A>
+
+<h2>You make a lot of releases of MaraDNS; at our
+    ISP/IT department, updating software is non-trivial.</h2>
+
+The number of releases seen in the <A href=changelog.html>changelog</A>
+is not an accurate reflection of how often someone using a stable
+branch of MaraDNS will need to update.
+
+<p>
+
+There were only three updates to the 1.0 legacy branch in 2006, and (so far)
+only two updates to the 1.0 branch in 2007.  The 1.2 branch
+was updated frequently in the first half of 2006, since I felt MaraDNS 1.2
+needed some features that didn't make it in to 1.2.00.  During this update
+cycle, there was always a stable bugfix-only branch of MaraDNS.
+
+<p>
+
+In August of 2006, I stabilized the 1.2 branch and only five updates 
+have been done since then.  Unless there is a critical bug, I only will update
+the 1.2 branch approximately once every six months or so.
+
+<p>
+
+I go to a great deal of effort to make sure MaraDNS releases are as
+painless to update as possible.  I ensure configuration file format
+compatibility, even between major versions of MaraDNS.  With the exception
+of configuration file parser bugfixes, MaraDNS 1.0 configuration files
+are compatible with MaraDNS 1.2 and 1.3.  
+
+<p>
+
+It is impossible to make code that is bug-free or without security 
+problems.  This is especially true with code that runs on the public
+internet.<sup><font size=-2>1</font></sup>  Code has to be updated from
+time to time.  What I do in order to minimize the disruption caused by an
+update is to always have a stable bugfix-only branch of MaraDNS (right now I 
+have <i>three</i> bugfix-only branches), and to, as much as possible, evenly
+space out the bugfix updates.
+
+<p>
+
+Footnote 1: Even DJB's code has security problems.  Both Qmail and DjbDNS
+have known security problems, and need to be patched before put on a public
+internet server.
 
