--- maradns-1.3.07.04/doc/en/man/Makefile	2006-08-14 15:02:07.000000000 -0500
+++ maradns-1.3.07.05/doc/en/man/Makefile	2007-09-03 13:34:09.000000000 -0500
@@ -1,12 +1,12 @@
 all: maradns.8 mararc.5 askmara.1 getzone.1 zoneserver.8 csv1.5 duende.8 \
-	csv2.5 csv2_txt.5 fetchzone.1 mqhash.1
+	csv2.5 csv2_txt.5 fetchzone.1 mqhash.1 bind2csv2.1
 
 ps: maradns.ps mararc.ps askmara.ps getzone.ps zoneserver.ps csv1.ps
 
 clean:
 	rm -f maradns.8 mararc.5 askmara.1 getzone.1 zoneserver.8 csv1.5 \
 	maradns.ps mararc.ps askmara.ps getzone.ps zoneserver.ps csv1.ps \
-	csv2.5 csv2_txt.5
+	csv2.5 csv2_txt.5 bind2csv2.1
 
 duende.8: ../source/duende.ej 
 	../../../tools/ej/ej2man ../source/duende.ej > duende.8
@@ -41,6 +41,9 @@
 mqhash.1: ../source/mqhash.ej
 	../../../tools/ej/ej2man ../source/mqhash.ej > mqhash.1
 
+bind2csv2.1: ../source/bind2csv2-man.ej
+	../../../tools/ej/ej2man ../source/bind2csv2-man.ej > bind2csv2.1
+
 maradns.ps: maradns.8
 	groff -mman -rS12 -Tps maradns.8 > maradns.ps
 
