--- maradns-1.3.07.06/tcp/zoneserver.c.orig	2007-10-18 10:17:25.000000000 -0500
+++ maradns-1.3.07.06/tcp/zoneserver.c	2007-10-18 10:17:59.000000000 -0500
@@ -1178,7 +1178,7 @@
              * */
 
             bind_addresses = libtcp_bind_address(1);
-            if(bind_addresses[0].ip == 0xffffffff) {
+            if(bind_addresses == 0 || bind_addresses[0].ip == 0xffffffff) {
                     harderror("Could not make list of bind addresses");
             }
             bind_address_iterate = 0;
