--- deadwood-3.2.01/src/DwUdpSocket.c	2012-02-11 14:15:43.000000000 -0600
+++ deadwood-3.2.02/src/DwUdpSocket.c	2012-02-12 11:25:57.000000000 -0600
@@ -813,6 +813,9 @@
         } else {
                 answer = dw_packet_to_cache(packet,count,is_nxdomain);
                 decomp = dwc_decompress(question,answer);
+		if(decomp == 0) {
+                        goto catch_cache_dns_reply;
+                }
                 if(dwc_has_bad_ip(decomp,blacklist_dict)) {
                         ret = -2; /* Tell caller we need synth "not there" */
                         goto catch_cache_dns_reply;
