--- deadwood-2.9.01/bug.reporting.policy	2010-07-22 10:08:25.000000000 -0700
+++ deadwood-2.9.02/bug.reporting.policy	2010-07-28 10:40:53.000000000 -0700
@@ -13,3 +13,8 @@
 	MaraDNS/Deadwood support for people not comfortable with mailing
 	lists, but no promises.
 
+* Bugs reporting Deadwood crashes will be marked "unreproducible" unless
+	either a stack trace (or core dump if the program was compiled with
+	'-g' to enable debugging symbols) or a recipe to reproduce the
+	crash is provided.
+
