--- deadwood-2.9.02/src/DwRecurse.c.orig	2010-07-28 10:45:01.000000000 -0700
+++ deadwood-2.9.02/src/DwRecurse.c	2010-07-28 10:56:46.000000000 -0700
@@ -1502,13 +1502,28 @@
 
         view = dwx_create_dns_details(in,query);
 
-        if(view == 0 ||
-           dwx_check_answer_section(in,query,view) == -1 ||
-           dwx_if_an_then_no_ns_nor_ar(view) == -1 ||
-           dwx_cleanup_ns_ar(view) == -1 ||
-           dwx_check_bailiwick_ns_section(view,query,bailiwick) == -1 ||
-           dwx_link_ns_records(view) == -1
-           ) {
+        if(view == 0) {
+		printf("view == 0\n");
+		goto catch_dwx_dissect_packet;
+	}
+	if(dwx_check_answer_section(in,query,view) == -1) {
+		printf("dwx_check_answer_section(in,query,view) == -1\n");
+		goto catch_dwx_dissect_packet;
+	}
+	if(dwx_if_an_then_no_ns_nor_ar(view) == -1) {
+		printf("dwx_if_an_then_no_ns_nor_ar(view) == -1\n");
+		goto catch_dwx_dissect_packet;
+	}
+	if(dwx_cleanup_ns_ar(view) == -1) {
+		printf("dwx_cleanup_ns_ar(view) == -1\n");
+		goto catch_dwx_dissect_packet;
+	}
+	if(dwx_check_bailiwick_ns_section(view,query,bailiwick) == -1) {
+		printf("dwx_check_bailiwick_ns_section()\n");
+		goto catch_dwx_dissect_packet;
+	}
+	if(dwx_link_ns_records(view) == -1) {
+		printf("dwx_link_ns_records(view) == -1\n");
                 goto catch_dwx_dissect_packet;
         }
         type = dwx_determine_answer_type(view,query);
