--- deadwood-2.9.02/src/DwRecurse.c.orig	2010-07-28 10:45:01.000000000 -0700
+++ deadwood-2.9.02/src/DwRecurse.c	2010-07-28 12:23:03.000000000 -0700
@@ -765,7 +765,8 @@
         for(ns = 0; ns < max * 2; ns += 2) { /* For each NS record */
                 /* Check its bailiwick */
                 if(dwx_string_in_bailiwick(in,look->ns[ns],bailiwick,
-                                query) != 1) {
+                                query) != 1 
+				&& view->ns_types[(ns >> 1)] != RR_SOA) {
                         view->ns_types[(ns >> 1)] = RRX_OOB_NS;
                 }
         }
