--- deadwood-2.9.02/src/DwRecurse.c.orig	2010-07-28 14:35:29.000000000 -0700
+++ deadwood-2.9.02/src/DwRecurse.c	2010-07-28 15:41:27.000000000 -0700
@@ -1434,7 +1434,7 @@
         if(view->ns_types[0] == RR_SOA) {
                 return TYPE_NOT_THERE; /* May be NXDOMAIN; look at header */
         }
-        if((view->ns_types[0] & RRX_NS_MASK) == RRX_GLUELESS_NS) {
+        if(view->ns_types[0] == RR_NS) {
                 return TYPE_NS_REFER;
         }
         return TYPE_NO_USEFUL_DATA;
@@ -1505,13 +1505,15 @@
         if(view == 0 ||
            dwx_check_answer_section(in,query,view) == -1 ||
            dwx_if_an_then_no_ns_nor_ar(view) == -1 ||
-           dwx_cleanup_ns_ar(view) == -1 ||
-           dwx_check_bailiwick_ns_section(view,query,bailiwick) == -1 ||
-           dwx_link_ns_records(view) == -1
-           ) {
+           dwx_cleanup_ns_ar(view) == -1) {
                 goto catch_dwx_dissect_packet;
         }
         type = dwx_determine_answer_type(view,query);
+	if(type == TYPE_NS_REFER && (
+           dwx_check_bailiwick_ns_section(view,query,bailiwick) == -1 ||
+           dwx_link_ns_records(view) == -1 )) {
+                goto catch_dwx_dissect_packet;
+        }
 
         out = dwx_make_cache_string(view,type);
         dwx_zap_dns_details(view);
