--- maradns-1.2.00/maradns.gpg.key	2002-07-19 00:03:22.000000000 -0700
+++ maradns-1.2.01/maradns.gpg.key	2005-12-31 21:13:29.000000000 -0800
@@ -1,11 +1,7 @@
 This is the GPG key I use for signing MaraDNS releases.
 
-Franky is also authoritzed to sign releases; his GPG key
-is the file named "franky.gpg.key".
-
-I go to some effort to keep this key secure; the private portion
-of it only exists on a laptop which has no public internet 
-services.
+This GPG key has existed since October 31, 2001 and hasn't
+changed since the 0.8.29 release of MaraDNS.
 
 -----BEGIN PGP PUBLIC KEY BLOCK-----
 Version: GnuPG v1.0.6 (GNU/Linux)
