--- deadwood-3.0.04/src/DwUdpSocket.c	2011-11-01 10:25:24.000000000 -0600
+++ deadwood-fix_reject_aaaa/src/DwUdpSocket.c	2011-11-07 23:48:31.067213930 -0600
@@ -690,6 +690,9 @@
                 /* Flag this as an answer */
                 answer[2] |= 0x80;
 
+		/* Flag RA bit because, well, recursion is available */
+		answer[3] |= 0x80;
+
                 /* One "NS" record; no other records */
                 answer[9] = 1;
                 answer[6] = answer[7] = answer[8] = answer[10] = answer[11] =0;
