--- deadwood-3.2.01/src/DwHash.c.orig	2012-01-15 10:45:09.518541087 -0600
+++ deadwood-3.2.01/src/DwHash.c	2012-01-15 10:51:52.778219757 -0600
@@ -389,7 +389,11 @@
                 return 0;
         }
 
-        if(use_fila == 1 && ignore_expire != 1 && get_time() > seek->expire
+	/* Check to see if a record has expired.  There are corner cases where 
+         * records can appear far in the future, so records which expire
+         * in over three years are considered already expired */
+        if(use_fila == 1 && ignore_expire != 1 && (get_time() > seek->expire 
+			|| (get_time() + 94608000) < seek->expire)
                         && seek->expire != 0) {
                 return 0;
         }
@@ -793,13 +797,15 @@
 void dwh_put_int64(FILE *handle, int64_t num) {
         int counter = 0;
         uint8_t val = 0;
-        /* If the number is negative, put 0x0000000000000000 in the file.
-         * No, Deadwood will not put an accurate timestamp before March
-         * 20, 1979 in the cache file.  If you wonder: Blake's 7 Gambit */
+        /* If the number is negative, put 0x0000000000000001 in the file.
+         * (not 0, which means "never expire") No, Deadwood will not put 
+         * an accurate timestamp before March 20, 1979 in the cache file.  
+         * If you wonder: Blake's 7 Gambit's original broadcast is our epoch */
         if(num < 0) {
-                for(counter = 0; counter < 8; counter++) {
+                for(counter = 0; counter < 7; counter++) {
                         putc(0,handle);
                 }
+                putc(1,handle);
                 return;
         }
         for(counter = 56; counter >= 0; counter -= 8) {
