--- deadwood-3.1.03/doc/Deadwood.ej	2011-12-21 13:44:37.000000000 -0600
+++ deadwood-3.2.01/doc/Deadwood.ej	2012-01-17 15:49:15.932749534 -0600
@@ -31,7 +31,7 @@
 <li>Ability to use expired entries in the cache when it is impossible to
     contact upstream DNS servers.
 
-<li>Ipv6 support can be compiled in if desired
+<li>IPv6 support can be compiled in if desired
 
 <li>Both DNS-over-UDP and DNS-over-TCP are handled by the same daemon
 
@@ -485,13 +485,13 @@
 
 <h1>ip/mask format of IPs</h1>
 Deadwood uses a standard ip/netmask formats to specify IPs.  
-An ip is in dotted-decimal format, e.g. "10.1.2.3" (or in ipv6
-format when ipv6 support is compiled in).
+An ip is in dotted-decimal format, e.g. "10.1.2.3" (or in IPv6
+format when IPv6 support is compiled in).
 <p>
 
 The netmask is used to specify a range of IPs.
 The netmask is a single number between
-1 and 32 (128 when ipv6 support is compiled in), which indicates the 
+1 and 32 (128 when IPv6 support is compiled in), which indicates the 
 number of leading "1" bits in the netmask.
 <p>
 
@@ -511,14 +511,28 @@
 a single IP will match.
 
 <h1>DNS over TCP</h1>
+DNS-over-TCP needs to be explicitly enabled by setting tcp_listen to 1.
+
+<p>
+
+Deadwood extracts useful information from UDP DNS packets marked truncated
+which almost always removes the need to have DNS-over-TCP.  However,
+Deadwood does not cache DNS packets larger than 512 bytes in size that
+need to be sent using TCP.  In addition, DNS-over-TCP packets which are
+"incomplete" DNS replies (replies which a stub resolver can not use,
+which can be either a NS referral or an incomplete CNAME reply) are not
+handled correctly by Deadwood.
+
+<p>
+
 Deadwood has support for both DNS-over-UDP and DNS-over-TCP; the same
-daemon listens on both the UDP and TCP DNS port.  DNS-over-TCP must
-be explicitly enabled by setting tcp_listen to 1.
+daemon listens on both the UDP and TCP DNS port.  
 
 <p>
 Only UDP DNS queries are cached.  Deadwood does not support caching
-over TCP; it handles TCP to resolve the occasional truncated reply or
-handle the occasional non-RFC-compliant TCP-only DNS resolver.
+over TCP; it handles TCP to resolve the rare truncated reply without any
+useful information or to work with very uncommon non-RFC-compliant TCP-only
+DNS resolvers.  In the real world, DNS-over-TCP is almost never used.
 
 <h1>Parsing other files</h1>
 
@@ -554,7 +568,7 @@
 
 This server can also be optionally compiled to have IPv6 support.  In order
 to enable IPv6 support, add '-DIPV6' to the compile-time flags.  For
-example, to compile this to make a small binary, and to have ipv6 support:
+example, to compile this to make a small binary, and to have IPv6 support:
 
 <pre>
 	export FLAGS='-Os -DIPV6'
@@ -688,15 +702,12 @@
 
 <h1>BUGS</h1>
 
-DNS-over-TCP needs to be explicitly enabled.  Note that DNS-over-TCP is 
-almost never used.  Also,
-Deadwood does not cache DNS packets larger than 512 bytes in size that
-need to be sent using TCP.  In addition, DNS-over-TCP packets which
-are "incomplete" DNS replies (replies which a stub resolver can not use,
-namely either a NS referral or an incomplete CNAME reply) are not
-handled correctly by Deadwood.  Again, DNS-over-TCP is very rare.
-<p>
+Deadwood does not follow RFC2181's advice to ignore DNS responses with the TC
+(truncated) bit set, but instead extracts the first RR.  If this is not 
+desired, set the undocumented parameter truncation_hack to 0 (but 
+read the DNS over TCP section of this man page).
 
+<p>
 Deadwood can not process DNS resource record types with numbers between
 65392 and 65407.  These RR types are marked by the IANA for "private use";
 Deadwood reserves these record types for internal use.  This is only 16
@@ -725,7 +736,7 @@
 <h1>AUTHORS</h1>
 Sam Trenholme (<A href=http://www.samiam.org/>http://www.samiam.org</a>) is 
 responsible for this program and man page.  He appreciates all of 
-Jean-Jacques Sarton's help giving this program Ipv6 support.
+Jean-Jacques Sarton's help giving this program IPv6 support.
 
 </BODY>
 
