--- deadwood-3.1.04/doc/FAQ.ej.orig	2012-01-09 10:21:36.207766044 -0600
+++ deadwood-3.1.04/doc/FAQ.ej	2012-01-09 10:24:40.382986080 -0600
@@ -6,13 +6,14 @@
 
 <h2>What is Deadwood?</h2>
 
-Deadwood is the recursive DNS daemon (service) for the up and coming
-MaraDNS 2.0.  MaraDNS 2.0 is going to have separate programs for
+Deadwood is the recursive DNS daemon (service) for
+MaraDNS 2.0.  MaraDNS 2.0 uses separate programs for
 authoritative records (maradns) and recursive records (Deadwood).  
 Deadwood is a standalone recursive server that can either be used
-in conjunction with MaraDNS, or by itself.  The program can run
-either in CentOS 5 (and hopefully other Linux and *NIX flavors) or
-in Windows XP (as well as newer Windows releases).
+in conjunction with MaraDNS's authoritative server, or by itself.
+The program can run either in Scientific Linux 6 (and hopefully other
+Linux and *NIX flavors) or in Windows XP (as well as newer Windows
+releases).
 
 <p>
 
@@ -27,7 +28,7 @@
 
 Ever since 2002, my plan has been to rewrite MaraDNS' recursive code.  
 In the fall of 2007, I finally started making the code; the code
-became feature complete in the summer of 2010.
+became MaraDNS' recursive resolver in the fall of 2010.
 
 <h2>How do I use Deadwood?</h2>
 
