--- deadwood-3.1.04/doc/FAQ.ej.orig	2012-01-11 15:36:27.011355774 -0600
+++ deadwood-3.1.04/doc/FAQ.ej	2012-01-11 15:48:35.506110079 -0600
@@ -4,6 +4,26 @@
 <BODY>
 <H1>Deadwood FAQ</H1>
 
+<h2>INDEX</h2>
+<BLOCKQUOTE>
+1. <A href="#01">What is Deadwood?</A><p>
+2. <A href="#02">How do I use Deadwood?</A><p>
+3. <A href="#03">How do I convert a MaraDNS mararc file in to a 
+                 Deadwood dwood3rc file?</A><p>
+4. <A href="#04">I changed a configuration parameter but it has not 
+		 affected Deadwood</A><p>
+5. <A href="#05">Deadwood sends out a lot of queries</A><p>
+6. <A href="#06">Steve Gibson's DNS benchmark reports that Deadwood 
+                 is dropping a lot of DNS packets</A><p>
+7. <A href="#07">Can Deadwood blacklist by domain?</A><p>
+8. <A href="#08">Does Deadwood have DNSSEC support?</A><p>
+9. <A href="#09">Records added to the cache when the timestamp is set
+                 to 1970 do not expire</A><p>
+10. <A href="#10">I get the error message "Unknown dwood3rc string 
+		 parameter"</A><p>
+</BLOCKQUOTE>
+
+<A name="01"> </A>
 <h2>What is Deadwood?</h2>
 
 Deadwood is the recursive DNS daemon (service) for
@@ -30,6 +50,7 @@
 In the fall of 2007, I finally started making the code; the code
 became MaraDNS' recursive resolver in the fall of 2010.
 
+<A name="02"> </A>
 <h2>How do I use Deadwood?</h2>
 
 Create a configuration file, <tt>/etc/dwood3rc</tt>, that looks like this:
@@ -45,6 +66,7 @@
 The above configuration file will only allow connections using the loopback
 interface on the same machine to resolve domains with Deadwood.
 
+<A name="03"> </A>
 <h2>How do I convert a MaraDNS mararc file in to a Deadwood dwood3rc file?</h2>
 
 While some effort has been made to have Deadwood use the same syntax and 
@@ -59,6 +81,7 @@
 
 </ul> 
 
+<A name="04"> </A>
 <h2>I changed a configuration parameter but it has not affected Deadwood</h2>
 
 Be sure to delete the cache file when making any changes to Deadwood's
@@ -66,6 +89,7 @@
 the dwood3rc.txt file is edited); in CentOS, with the default dwood3rc file,
 the file is called dw_cache.
 
+<A name="05"> </A>
 <h2>Deadwood sends out a lot of queries</h2>
 
 Deadwood will do this on a slow network, since the default parameters are
@@ -79,6 +103,7 @@
 num_retries = 1
 </pre>
 
+<A name="06"> </A>
 <h2>Steve Gibson's DNS benchmark reports that Deadwood is dropping a lot of 
     DNS packets</h2>
 
@@ -91,6 +116,7 @@
 drops very few (if any) DNS packets sent to it.  Namebench is available at
 available at http://code.google.com/p/namebench/
 
+<A name="07"> </A>
 <h2>Can Deadwood blacklist by domain?</h2>
 
 Yes.
@@ -125,12 +151,14 @@
 If it is more convenient to store the domains in separate files, this can be
 done using Deadwood's "execfile" mechanism.
 
+<A name="08"> </A>
 <h2>Does Deadwood have DNSSEC support?</h2>
 
 No.  I have nothing against DNSSEC per se, but I plain simply am not in a
 position to take the time and effort to implement DNSSEC without being
 compensated for my work.
 
+<A name="09"> </A>
 <h2>Records added to the cache when the timestamp is set to 1970 do not
 expire</h2>
 
@@ -183,6 +211,7 @@
 </li>
 </ul>
 
+<A name="10"> </A>
 <h2>I get the error message "Unknown dwood3rc string parameter"</h2>
 
 This error message indicates either one of two things:
